// @workday/canvas-kit-react-modal/dist/commonjs/lib/Modal.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var canvas_kit_react_popup_1 = require("@workday/canvas-kit-react-popup");
var ModalContent_1 = __importDefault(require("./ModalContent"));
var ModalWidth;
(function (ModalWidth) {
    ModalWidth["s"] = "440px";
    ModalWidth["m"] = "800px";
})(ModalWidth = exports.ModalWidth || (exports.ModalWidth = {}));
var Modal = function (_a) {
    var _b = _a.open, open = _b === void 0 ? false : _b, _c = _a.padding, padding = _c === void 0 ? canvas_kit_react_popup_1.PopupPadding.l : _c, _d = _a.width, width = _d === void 0 ? ModalWidth.s : _d, container = _a.container, modalContentProps = __rest(_a, ["open", "padding", "width", "container"]);
    return open && typeof window !== 'undefined' ? (React.createElement(ModalContent_1.default, __assign({ container: container, padding: padding, width: width }, modalContentProps))) : null;
};
Modal.Padding = canvas_kit_react_popup_1.PopupPadding;
Modal.Width = ModalWidth;
exports.default = Modal;
function useModal() {
    var _a = React.useState(false), open = _a[0], setOpen = _a[1];
    return {
        targetProps: {
            onClick: function () {
                setOpen(true);
            },
        },
        closeModal: function () {
            setOpen(false);
        },
        modalProps: {
            open: open,
            handleClose: function () {
                setOpen(false);
            },
        },
    };
}
exports.useModal = useModal;
