// @workday/canvas-kit-react-color-picker/dist/commonjs/lib/parts/ColorSwatch.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var canvas_kit_react_common_1 = require("@workday/canvas-kit-react-common");
var React = __importStar(require("react"));
var styled_1 = __importDefault(require("@emotion/styled"));
var canvas_system_icons_web_1 = require("@workday/canvas-system-icons-web");
var canvas_kit_react_icon_1 = require("@workday/canvas-kit-react-icon");
var Container = styled_1.default('div')({
    width: 20,
    height: 20,
    borderRadius: canvas_kit_react_core_1.borderRadius.s,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    '& > *': {
        margin: -2,
    },
}, function (_a) {
    var color = _a.color, showCheck = _a.showCheck;
    return ({
        backgroundColor: color,
        boxShadow: showCheck || color === canvas_kit_react_core_1.colors.frenchVanilla100
            ? 'inset 0px 0px 0px 1px rgba(0, 0, 0, 0.25)'
            : undefined,
    });
});
exports.ColorSwatch = function (_a) {
    var color = _a.color, _b = _a.showCheck, showCheck = _b === void 0 ? false : _b, elemProps = __rest(_a, ["color", "showCheck"]);
    return (React.createElement(Container, __assign({ color: color, showCheck: showCheck }, elemProps), showCheck && (React.createElement(canvas_kit_react_icon_1.SystemIcon, { fill: canvas_kit_react_common_1.pickForegroundColor(color), fillHover: canvas_kit_react_common_1.pickForegroundColor(color), icon: canvas_system_icons_web_1.checkSmallIcon, color: color }))));
};
