// @workday/canvas-kit-react-form-field/dist/commonjs/lib/Label.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var styled_1 = __importDefault(require("@emotion/styled"));
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var canvas_kit_react_common_1 = require("@workday/canvas-kit-react-common");
var types_1 = require("./types");
var labelStyles = [
    __assign(__assign(__assign({}, canvas_kit_react_core_1.type.body), canvas_kit_react_core_1.type.variant.label), { padding: 0 }),
    function (_a) {
        var labelPosition = _a.labelPosition, accessibleHide = _a.accessibleHide;
        if (accessibleHide) {
            return canvas_kit_react_common_1.accessibleHide;
        }
        return __assign({}, (labelPosition === types_1.FormFieldLabelPosition.Left
            ? {
                marginRight: canvas_kit_react_core_1.spacing.l,
                minWidth: 180,
                display: 'flex',
                alignItems: 'center',
                maxHeight: canvas_kit_react_core_1.spacing.xl,
            }
            : {
                display: 'block',
                marginBottom: canvas_kit_react_core_1.spacing.xxxs,
            }));
    },
];
var RequiredAstrisk = styled_1.default('abbr')({
    color: canvas_kit_react_core_1.colors.cinnamon500,
    fontSize: '16px',
    fontWeight: 400,
    top: '1px',
    paddingLeft: '2px',
    textDecoration: 'unset',
});
var LegendComponent = styled_1.default('legend').apply(void 0, labelStyles);
var LabelComponent = styled_1.default('label').apply(void 0, labelStyles);
var Label = (function (_super) {
    __extends(Label, _super);
    function Label() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Label.prototype.render = function () {
        var _a = this.props, _b = _a.labelPosition, labelPosition = _b === void 0 ? Label.Position.Top : _b, _c = _a.isLegend, isLegend = _c === void 0 ? false : _c, _d = _a.requiredLabel, requiredLabel = _d === void 0 ? 'required' : _d, elemProps = __rest(_a, ["labelPosition", "isLegend", "requiredLabel"]);
        var children = !this.props.required
            ? this.props.children
            : [
                this.props.children,
                React.createElement(RequiredAstrisk, { key: '0', title: requiredLabel }, "*"),
            ];
        return (React.createElement(React.Fragment, null, isLegend ? (React.createElement(LegendComponent, __assign({ labelPosition: labelPosition, isLegend: isLegend }, elemProps, { children: children }))) : (React.createElement(LabelComponent, __assign({ labelPosition: labelPosition }, elemProps, { children: children })))));
    };
    Label.Position = types_1.FormFieldLabelPosition;
    return Label;
}(React.Component));
exports.default = Label;
