// @storybook/design-system/dist/components/tooltip/ListItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListItem = ListItem;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styles = require("../shared/styles");

var _animation = require("../shared/animation");

function _templateObject12() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      cursor: not-allowed !important;\n      ", ", ", " {\n        color: ", ";\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", " {\n        ", ";\n        flex: 0 1 auto;\n        display: inline-block;\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", " {\n        font-weight: ", ";\n      }\n      ", ", ", " {\n        color: ", ";\n      }\n\n      ", " svg {\n        opacity: 1;\n        path {\n          fill: ", ";\n        }\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-size: ", "px;\n  transition: all 150ms ease-out;\n  color: ", ";\n  text-decoration: none;\n  display: block;\n\n  /* Styling */\n  ", " {\n    color: ", ";\n  }\n\n  ", " svg {\n    transition: all 200ms ease-out;\n    opacity: 0;\n    height: 12px;\n    width: 12px;\n    margin: 3px 0;\n    vertical-align: top;\n\n    path {\n      fill: ", ";\n    }\n  }\n\n  &:hover {\n    background: #e3f3ff;\n    cursor: pointer;\n\n    ", " svg {\n      opacity: 1;\n    }\n  }\n\n  ", ";\n\n  ", ";\n\n  ", ";\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  /* Layout */\n  line-height: 18px;\n  padding: 7px 15px;\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n\n  ", " {\n    display: block;\n  }\n\n  ", ", ", ", ", " {\n    display: inline-flex;\n  }\n\n  ", "{\n    flex: 0 1 auto;\n    margin-right: auto;\n  }\n\n  ", " {\n    flex: 0 1 auto;\n    margin-left: auto;\n    margin-right: auto;\n  }\n\n  ", ", ", " {\n    flex: 0 1 auto;\n  }\n\n  ", " {\n    flex: none;\n    text-align: right;\n    margin-left: 10px;\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  list-style: none;\n\n  &:not(:first-child) {\n    border-top: 1px solid ", ";\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-weight: ", ";\n  white-space: nowrap;\n  overflow: hidden;\n  text-overflow: ellipsis;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Left = _styledComponents["default"].span(_templateObject());

var Title = _styledComponents["default"].span(_templateObject2(), _styles.typography.weight.bold);

var Center = _styledComponents["default"].span(_templateObject3());

var Right = _styledComponents["default"].span(_templateObject4());

var ItemWrapper = _styledComponents["default"].li(_templateObject5(), _styles.color.mediumlight);

var ItemInner = _styledComponents["default"].span(_templateObject6(), Title, Left, Center, Right, Title, Center, Left, Right, Right);

var linkStyles = (0, _styledComponents.css)(_templateObject7(), _styles.typography.size.s1, _styles.color.mediumdark, Title, _styles.color.darker, Right, _styles.color.mediumdark, Right, function (props) {
  return props.active && (0, _styledComponents.css)(_templateObject8(), Title, _styles.typography.weight.bold, Title, Center, props.activeColor, Right, props.activeColor);
}, function (props) {
  return props.isLoading && (0, _styledComponents.css)(_templateObject9(), Title, _animation.inlineGlow);
}, function (props) {
  return props.disabled && (0, _styledComponents.css)(_templateObject10(), Title, Center, _styles.color.mediumdark);
}); // eslint-disable-next-line jsx-a11y/anchor-has-content

var Item = (0, _styledComponents["default"])(function (_ref) {
  var active = _ref.active,
      activeColor = _ref.activeColor,
      isLoading = _ref.isLoading,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["active", "activeColor", "isLoading"]);
  return /*#__PURE__*/_react["default"].createElement("a", rest);
})(_templateObject11(), linkStyles);

var buildStyledLinkWrapper = function buildStyledLinkWrapper(LinkWrapper) {
  return (0, _styledComponents["default"])(function (_ref2) {
    var active = _ref2.active,
        isLoading = _ref2.isLoading,
        activeColor = _ref2.activeColor,
        linkWrapperRest = (0, _objectWithoutProperties2["default"])(_ref2, ["active", "isLoading", "activeColor"]);
    return /*#__PURE__*/_react["default"].createElement(LinkWrapper, linkWrapperRest);
  })(_templateObject12(), linkStyles);
};

function ListItem(_ref3) {
  var appearance = _ref3.appearance,
      left = _ref3.left,
      title = _ref3.title,
      center = _ref3.center,
      right = _ref3.right,
      onClick = _ref3.onClick,
      LinkWrapper = _ref3.LinkWrapper,
      rest = (0, _objectWithoutProperties2["default"])(_ref3, ["appearance", "left", "title", "center", "right", "onClick", "LinkWrapper"]);
  var listItemActiveColor = _styles.color[appearance];

  var linkInner = /*#__PURE__*/_react["default"].createElement(ItemInner, {
    onClick: onClick,
    role: "presentation"
  }, left && /*#__PURE__*/_react["default"].createElement(Left, null, left), title && /*#__PURE__*/_react["default"].createElement(Title, null, title), center && /*#__PURE__*/_react["default"].createElement(Center, null, center), right && /*#__PURE__*/_react["default"].createElement(Right, null, right));

  if (LinkWrapper) {
    var StyledLinkWrapper = buildStyledLinkWrapper(LinkWrapper);
    return /*#__PURE__*/_react["default"].createElement(ItemWrapper, null, /*#__PURE__*/_react["default"].createElement(StyledLinkWrapper, (0, _extends2["default"])({
      activeColor: listItemActiveColor
    }, rest), linkInner));
  }

  return /*#__PURE__*/_react["default"].createElement(ItemWrapper, null, /*#__PURE__*/_react["default"].createElement(Item, (0, _extends2["default"])({
    activeColor: listItemActiveColor
  }, rest), linkInner));
}

ListItem.propTypes = {
  appearance: _propTypes["default"].oneOf(['primary', 'secondary']),
  isLoading: _propTypes["default"].bool,
  left: _propTypes["default"].node,
  title: _propTypes["default"].node,
  center: _propTypes["default"].node,
  right: _propTypes["default"].node,
  active: _propTypes["default"].bool,
  disabled: _propTypes["default"].bool,
  LinkWrapper: _propTypes["default"].func,
  onClick: _propTypes["default"].func
};
ListItem.defaultProps = {
  appearance: 'primary',
  isLoading: false,
  left: null,
  title: /*#__PURE__*/_react["default"].createElement("span", null, "Loading"),
  center: null,
  right: null,
  active: false,
  disabled: false,
  LinkWrapper: undefined,
  onClick: undefined
};
