// @workday/canvas-kit-react-radio/dist/es6/lib/Radio.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { focusRing, mouseFocusBehavior, styled, useUniqueId, } from '@workday/canvas-kit-react-common';
import canvas, { borderRadius, colors, inputColors, spacingNumbers as spacing, } from '@workday/canvas-kit-react-core';
var radioBorderRadius = 9;
var radioDot = 8;
var radioHeight = 18;
var radioTapArea = spacing.m;
var radioContainerHeight = radioTapArea;
var radioLabelDistance = spacing.xs;
var radioWidth = 18;
var rippleRadius = (spacing.l - radioWidth) / 2;
var RadioContainer = styled('div')({
    display: 'flex',
    alignItems: 'center',
    minHeight: radioContainerHeight,
    position: 'relative',
});
var RadioInputWrapper = styled('div')({
    display: 'flex',
    height: radioHeight,
    width: radioWidth,
});
var RadioRipple = styled('span')({
    borderRadius: borderRadius.circle,
    boxShadow: "0 0 0 0 " + colors.soap200,
    height: radioHeight,
    transition: 'box-shadow 150ms ease-out',
    width: radioWidth,
    position: 'absolute',
    pointerEvents: 'none',
    zIndex: -1,
});
var RadioInput = styled('input')({
    borderRadius: radioBorderRadius,
    width: radioTapArea,
    height: radioTapArea,
    margin: 0,
    marginTop: '-3px',
    marginLeft: '-3px',
    position: 'absolute',
    opacity: 0,
    '&:focus, &:active': {
        outline: 'none',
    },
}, function (_a) {
    var checked = _a.checked, disabled = _a.disabled, _b = _a.theme.canvas.palette, themePrimary = _b.primary, themeFocusOutline = _b.common.focusOutline;
    return (__assign({ cursor: disabled ? undefined : 'pointer', '&:hover ~ span:first-of-type': {
            boxShadow: disabled ? undefined : "0 0 0 " + rippleRadius + "px " + colors.soap200,
        }, '&:hover ~ div:first-of-type': {
            backgroundColor: checked
                ? themePrimary.main
                : disabled
                    ? inputColors.disabled.background
                    : 'white',
            borderColor: checked
                ? themePrimary.main
                : disabled
                    ? inputColors.disabled.border
                    : inputColors.hoverBorder,
            borderWidth: '1px',
        }, '&:focus, &focus:hover': {
            '& ~ div:first-of-type': {
                borderColor: checked ? themePrimary.main : themeFocusOutline,
                borderWidth: '2px',
            },
        }, '&:checked:focus ~ div:first-of-type': __assign({}, focusRing({ separation: 2, outerColor: themeFocusOutline })) }, mouseFocusBehavior({
        '&:focus ~ div:first-of-type': __assign(__assign({}, focusRing({ width: 0, outerColor: themeFocusOutline })), { borderWidth: '1px', borderColor: checked ? themePrimary.main : inputColors.border }),
        '&:focus:hover ~ div:first-of-type, &:focus:active ~ div:first-of-type': {
            borderColor: checked ? themePrimary.main : inputColors.hoverBorder,
        },
    })));
});
var RadioBackground = styled('div')({
    alignItems: 'center',
    backgroundColor: colors.frenchVanilla100,
    borderRadius: radioBorderRadius,
    borderStyle: 'solid',
    borderWidth: '1px',
    boxSizing: 'border-box',
    display: 'flex',
    height: radioHeight,
    justifyContent: 'center',
    padding: '0px 2px',
    pointerEvents: 'none',
    position: 'absolute',
    transition: 'border 200ms ease, background 200ms',
    width: radioWidth,
}, function (_a) {
    var checked = _a.checked, disabled = _a.disabled, themePrimary = _a.theme.canvas.palette.primary;
    return ({
        borderColor: checked
            ? themePrimary.main
            : disabled
                ? inputColors.disabled.border
                : inputColors.border,
        backgroundColor: checked
            ? themePrimary.main
            : disabled
                ? inputColors.disabled.background
                : 'white',
    });
});
var RadioCheck = styled('div')({
    borderRadius: radioBorderRadius,
    display: 'flex',
    flexDirection: 'column',
    height: radioDot,
    pointerEvents: 'none',
    transition: 'transform 200ms ease, opacity 200ms ease',
    width: radioDot,
}, function (_a) {
    var theme = _a.theme;
    return ({
        backgroundColor: theme.canvas.palette.primary.contrast,
    });
}, function (_a) {
    var checked = _a.checked;
    return ({
        opacity: checked ? 1 : 0,
        transform: checked ? 'scale(1)' : 'scale(0.5)',
    });
});
var RadioLabel = styled('label')(__assign(__assign({}, canvas.type.body), { paddingLeft: radioLabelDistance }), function (_a) {
    var disabled = _a.disabled;
    return (disabled ? { color: inputColors.disabled.text } : { cursor: 'pointer' });
});
export var Radio = function (_a) {
    var _b = _a.checked, checked = _b === void 0 ? false : _b, id = _a.id, _c = _a.label, label = _c === void 0 ? '' : _c, disabled = _a.disabled, inputRef = _a.inputRef, name = _a.name, onChange = _a.onChange, value = _a.value, elemProps = __rest(_a, ["checked", "id", "label", "disabled", "inputRef", "name", "onChange", "value"]);
    var inputId = useUniqueId(id);
    return (React.createElement(RadioContainer, null,
        React.createElement(RadioInputWrapper, { disabled: disabled },
            React.createElement(RadioInput, __assign({ checked: checked, disabled: disabled, id: inputId, ref: inputRef, name: name, onChange: onChange, type: "radio", value: value, "aria-checked": checked }, elemProps)),
            React.createElement(RadioRipple, null),
            React.createElement(RadioBackground, { checked: checked, disabled: disabled },
                React.createElement(RadioCheck, { checked: checked }))),
        label && (React.createElement(RadioLabel, { htmlFor: inputId, disabled: disabled }, label))));
};
export default Radio;
