// @workday/canvas-kit-react-text-input/dist/commonjs/lib/TextInput.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var canvas_kit_react_common_1 = require("@workday/canvas-kit-react-common");
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var Input = canvas_kit_react_common_1.styled('input')(__assign(__assign({}, canvas_kit_react_core_1.type.body), { border: "1px solid " + canvas_kit_react_core_1.inputColors.border, display: 'block', backgroundColor: canvas_kit_react_core_1.inputColors.background, borderRadius: canvas_kit_react_core_1.borderRadius.m, boxSizing: 'border-box', height: 40, transition: '0.2s box-shadow, 0.2s border-color', padding: canvas_kit_react_core_1.spacingNumbers.xxs, margin: 0, '&::placeholder': {
        color: canvas_kit_react_core_1.inputColors.placeholder,
    }, '&:hover': {
        borderColor: canvas_kit_react_core_1.inputColors.hoverBorder,
    }, '&:focus:not([disabled])': {
        borderColor: canvas_kit_react_core_1.inputColors.focusBorder,
        boxShadow: "inset 0 0 0 1px " + canvas_kit_react_core_1.inputColors.focusBorder,
        outline: 'none',
    }, '&:disabled': {
        backgroundColor: canvas_kit_react_core_1.inputColors.disabled.background,
        borderColor: canvas_kit_react_core_1.inputColors.disabled.border,
        color: canvas_kit_react_core_1.inputColors.disabled.text,
        '&::placeholder': {
            color: canvas_kit_react_core_1.inputColors.disabled.text,
        },
    }, '::-ms-clear': {
        display: 'none',
    } }), function (_a) {
    var width = _a.width;
    return ({
        minWidth: width || 280,
        width: width,
    });
}, function (_a) {
    var grow = _a.grow;
    return grow && {
        width: '100%',
    };
}, function (_a) {
    var theme = _a.theme, error = _a.error;
    return __assign({ '&:focus:not([disabled])': {
            borderColor: theme.canvas.palette.common.focusOutline,
            boxShadow: "inset 0 0 0 1px " + theme.canvas.palette.common.focusOutline,
            outline: 'none',
        } }, canvas_kit_react_common_1.errorRing(error, theme));
});
var TextInput = (function (_super) {
    __extends(TextInput, _super);
    function TextInput() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    TextInput.prototype.render = function () {
        var _a = this.props, inputRef = _a.inputRef, grow = _a.grow, error = _a.error, inputProps = __rest(_a, ["inputRef", "grow", "error"]);
        return React.createElement(Input, __assign({ type: "text", ref: inputRef, grow: grow, error: error }, inputProps));
    };
    TextInput.ErrorType = canvas_kit_react_common_1.ErrorType;
    return TextInput;
}(React.Component));
exports.default = TextInput;
