// @modulz/radix-system/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var core = require('@styled-system/core');
var css = _interopDefault(require('@styled-system/css'));

var config = {
    fontSize: {
        property: 'fontSize',
        scale: 'fontSizes',
    },
};
var fontSize = core.system(config);

var config$1 = { fontStyle: true };
var fontStyle = core.system(config$1);

var config$2 = {
    fontFamily: {
        property: 'fontFamily',
        scale: 'fonts',
    },
};
var fontFamily = core.system(config$2);

var config$3 = {
    fontWeight: {
        property: 'fontWeight',
        scale: 'fontWeights',
    },
};
var fontWeight = core.system(config$3);

var config$4 = {
    fill: {
        property: 'fill',
        scale: 'colors',
    },
};
var fill = core.system(config$4);

var config$5 = {
    lineHeight: {
        property: 'lineHeight',
        scale: 'lineHeights',
    },
};
var lineHeight = core.system(config$5);

var config$6 = { textAlign: true };
var textAlign = core.system(config$6);

var config$7 = { display: true };
var display = core.system(config$7);

var config$8 = {
    height: {
        property: 'height',
        scale: 'sizes',
    },
};
var height = core.system(config$8);

var config$9 = {
    maxWidth: {
        property: 'maxWidth',
        scale: 'sizes',
    },
};
var maxWidth = core.system(config$9);

var config$a = {
    maxHeight: {
        property: 'maxHeight',
        scale: 'sizes',
    },
};
var maxHeight = core.system(config$a);

var config$b = {
    minHeight: {
        property: 'minHeight',
        scale: 'sizes',
    },
};
var minHeight = core.system(config$b);

var config$c = {
    minWidth: {
        property: 'minWidth',
        scale: 'sizes',
    },
};
var minWidth = core.system(config$c);

var config$d = {
    width: {
        property: 'width',
        scale: 'sizes',
    },
};
var width = core.system(config$d);

var config$e = { overflow: true };
var overflow = core.system(config$e);

var config$f = {
    border: {
        property: 'border',
        scale: 'borders',
    },
    borderWidth: {
        property: 'borderWidth',
        scale: 'borderWidths',
    },
    borderStyle: {
        property: 'borderStyle',
        scale: 'borderStyles',
    },
    borderColor: {
        property: 'borderColor',
        scale: 'colors',
    },
    borderTop: {
        property: 'borderTop',
        scale: 'borders',
    },
    borderRight: {
        property: 'borderRight',
        scale: 'borders',
    },
    borderBottom: {
        property: 'borderBottom',
        scale: 'borders',
    },
    borderLeft: {
        property: 'borderLeft',
        scale: 'borders',
    },
    borderX: {
        properties: ['borderLeft', 'borderRight'],
        scale: 'borders',
    },
    borderY: {
        properties: ['borderTop', 'borderBottom'],
        scale: 'borders',
    },
};
var border = core.system(config$f);

var config$g = {
    borderRadius: {
        property: 'borderRadius',
        scale: 'radii',
    },
    borderTopLeftRadius: {
        property: 'borderTopLeftRadius',
        scale: 'radii',
    },
    borderTopRightRadius: {
        property: 'borderTopRightRadius',
        scale: 'radii',
    },
    borderBottomLeftRadius: {
        property: 'borderBottomLeftRadius',
        scale: 'radii',
    },
    borderBottomRightRadius: {
        property: 'borderBottomRightRadius',
        scale: 'radii',
    },
};
var borderRadius = core.system(config$g);

var config$h = {
    boxShadow: {
        property: 'boxShadow',
        scale: 'shadows',
    },
};
var boxShadow = core.system(config$h);

var config$i = {
    opacity: true,
};
var opacity = core.system(config$i);

var config$j = {
    backgroundColor: {
        property: 'backgroundColor',
        scale: 'colors',
    },
};
config$j.bg = config$j.backgroundColor;
var backgroundColor = core.system(config$j);

var config$k = {
    textColor: {
        property: 'color',
        scale: 'colors',
    },
};
var textColor = core.system(config$k);

/*! *****************************************************************************
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the Apache License, Version 2.0 (the "License"); you may not use
this file except in compliance with the License. You may obtain a copy of the
License at http://www.apache.org/licenses/LICENSE-2.0

THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
MERCHANTABLITY OR NON-INFRINGEMENT.

See the Apache Version 2.0 License for specific language governing permissions
and limitations under the License.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

var isNumber = function (n) { return typeof n === 'number' && !isNaN(n); };
var getNegativeSpace = function (n, scale) {
    if (!isNumber(n)) {
        return core.get(scale, n, n);
    }
    var isNegative = n < 0;
    var absolute = Math.abs(n);
    var value = core.get(scale, absolute, absolute);
    if (!isNumber(value)) {
        return isNegative ? '-' + value : value;
    }
    return value * (isNegative ? -1 : 1);
};

var config$l = {
    margin: {
        property: 'margin',
        scale: 'space',
        transform: getNegativeSpace,
    },
    marginTop: {
        property: 'marginTop',
        scale: 'space',
        transform: getNegativeSpace,
    },
    marginRight: {
        property: 'marginRight',
        scale: 'space',
        transform: getNegativeSpace,
    },
    marginBottom: {
        property: 'marginBottom',
        scale: 'space',
        transform: getNegativeSpace,
    },
    marginLeft: {
        property: 'marginLeft',
        scale: 'space',
        transform: getNegativeSpace,
    },
    marginX: {
        properties: ['marginLeft', 'marginRight'],
        scale: 'space',
        transform: getNegativeSpace,
    },
    marginY: {
        properties: ['marginTop', 'marginBottom'],
        scale: 'space',
        transform: getNegativeSpace,
    },
};
config$l.m = config$l.margin;
config$l.mt = config$l.marginTop;
config$l.mr = config$l.marginRight;
config$l.mb = config$l.marginBottom;
config$l.ml = config$l.marginLeft;
config$l.mx = config$l.marginX;
config$l.my = config$l.marginY;
var margin = core.system(config$l);

var config$m = {
    padding: {
        property: 'padding',
        scale: 'space',
    },
    paddingTop: {
        property: 'paddingTop',
        scale: 'space',
    },
    paddingRight: {
        property: 'paddingRight',
        scale: 'space',
    },
    paddingBottom: {
        property: 'paddingBottom',
        scale: 'space',
    },
    paddingLeft: {
        property: 'paddingLeft',
        scale: 'space',
    },
    paddingX: {
        properties: ['paddingLeft', 'paddingRight'],
        scale: 'space',
    },
    paddingY: {
        properties: ['paddingTop', 'paddingBottom'],
        scale: 'space',
    },
};
config$m.p = config$m.padding;
config$m.pt = config$m.paddingTop;
config$m.pr = config$m.paddingRight;
config$m.pb = config$m.paddingBottom;
config$m.pl = config$m.paddingLeft;
config$m.px = config$m.paddingX;
config$m.py = config$m.paddingY;
var padding = core.system(config$m);

var spaceSet = core.system(__assign(__assign({}, config$l), config$m));

var config$n = { alignContent: true };
var alignContent = core.system(config$n);

var config$o = { alignItems: true };
var alignItems = core.system(config$o);

var config$p = { flexDirection: true };
var flexDirection = core.system(config$p);

var config$q = { flexWrap: true };
var flexWrap = core.system(config$q);

var config$r = { justifyContent: true };
var justifyContent = core.system(config$r);

var config$s = { justifyItems: true };
var justifyItems = core.system(config$s);

var flexContainerSet = core.system(__assign(__assign(__assign(__assign(__assign(__assign({}, config$n), config$o), config$p), config$q), config$r), config$s));

var config$t = { flex: true };
var flex = core.system(config$t);

var config$u = { flexGrow: true };
var flexGrow = core.system(config$u);

var config$v = { flexShrink: true };
var flexShrink = core.system(config$v);

var config$w = { flexBasis: true };
var flexBasis = core.system(config$w);

var config$x = { alignSelf: true };
var alignSelf = core.system(config$x);

var config$y = { justifySelf: true };
var justifySelf = core.system(config$y);

var config$z = { order: true };
var order = core.system(config$z);

var flexItemSet = core.system(__assign(__assign(__assign(__assign(__assign(__assign(__assign({}, config$t), config$u), config$v), config$w), config$x), config$y), config$z));

var config$A = { position: true };
var position = core.system(config$A);

var config$B = {
    zIndex: {
        property: 'zIndex',
        scale: 'zIndices',
    },
};
var zIndex = core.system(config$B);

var config$C = {
    top: {
        property: 'top',
        scale: 'space',
        transform: getNegativeSpace,
    },
};
var top = core.system(config$C);

var config$D = {
    right: {
        property: 'right',
        scale: 'space',
        transform: getNegativeSpace,
    },
};
var right = core.system(config$D);

var config$E = {
    bottom: {
        property: 'bottom',
        scale: 'space',
        transform: getNegativeSpace,
    },
};
var bottom = core.system(config$E);

var config$F = {
    left: {
        property: 'left',
        scale: 'space',
        transform: getNegativeSpace,
    },
};
var left = core.system(config$F);

var positionSet = core.system(__assign(__assign(__assign(__assign(__assign(__assign({}, config$A), config$B), config$C), config$D), config$E), config$F));

var config$G = {
    gridGap: {
        property: 'gridGap',
        scale: 'space',
    },
};
var gridGap = core.system(config$G);

var config$H = {
    gridColumnGap: {
        property: 'gridColumnGap',
        scale: 'space',
    },
};
var gridColumnGap = core.system(config$H);

var config$I = {
    gridRowGap: {
        property: 'gridRowGap',
        scale: 'space',
    },
};
var gridRowGap = core.system(config$I);

var config$J = { gridColumn: true };
var gridColumn = core.system(config$J);

var config$K = { gridRow: true };
var gridRow = core.system(config$K);

var config$L = { gridAutoFlow: true };
var gridAutoFlow = core.system(config$L);

var config$M = { gridAutoColumns: true };
var gridAutoColumns = core.system(config$M);

var config$N = { gridAutoRows: true };
var gridAutoRows = core.system(config$N);

var config$O = { gridTemplateColumns: true };
var gridTemplateColumns = core.system(config$O);

var config$P = { gridTemplateRows: true };
var gridTemplateRows = core.system(config$P);

var config$Q = { gridTemplateAreas: true };
var gridTemplateAreas = core.system(config$Q);

var config$R = { gridArea: true };
var gridArea = core.system(config$R);

var gridSet = core.system(__assign(__assign(__assign(__assign(__assign(__assign(__assign(__assign(__assign(__assign(__assign(__assign({}, config$G), config$H), config$I), config$J), config$K), config$L), config$M), config$N), config$O), config$P), config$Q), config$R));

var variant = function (config) { return function (props) {
    var _a;
    var prop = Object.keys(config)[0];
    // @ts-ignore
    var getValue = function (n, scale) { return core.get(scale, n); };
    getValue.defaults = config[prop];
    return css(core.createParser((_a = {},
        _a[prop] = getValue,
        _a))(props));
}; };

var sx = function (_a) {
    var sx = _a.sx, theme = _a.theme;
    return css(sx)(theme);
};

Object.defineProperty(exports, 'compose', {
  enumerable: true,
  get: function () {
    return core.compose;
  }
});
Object.defineProperty(exports, 'createParser', {
  enumerable: true,
  get: function () {
    return core.createParser;
  }
});
Object.defineProperty(exports, 'get', {
  enumerable: true,
  get: function () {
    return core.get;
  }
});
Object.defineProperty(exports, 'system', {
  enumerable: true,
  get: function () {
    return core.system;
  }
});
exports.alignContent = alignContent;
exports.alignItems = alignItems;
exports.alignSelf = alignSelf;
exports.backgroundColor = backgroundColor;
exports.border = border;
exports.borderRadius = borderRadius;
exports.bottom = bottom;
exports.boxShadow = boxShadow;
exports.display = display;
exports.fill = fill;
exports.flex = flex;
exports.flexBasis = flexBasis;
exports.flexContainerSet = flexContainerSet;
exports.flexDirection = flexDirection;
exports.flexGrow = flexGrow;
exports.flexItemSet = flexItemSet;
exports.flexShrink = flexShrink;
exports.flexWrap = flexWrap;
exports.fontFamily = fontFamily;
exports.fontSize = fontSize;
exports.fontStyle = fontStyle;
exports.fontWeight = fontWeight;
exports.gridArea = gridArea;
exports.gridAutoColumns = gridAutoColumns;
exports.gridAutoFlow = gridAutoFlow;
exports.gridAutoRows = gridAutoRows;
exports.gridColumn = gridColumn;
exports.gridColumnGap = gridColumnGap;
exports.gridGap = gridGap;
exports.gridRow = gridRow;
exports.gridRowGap = gridRowGap;
exports.gridSet = gridSet;
exports.gridTemplateAreas = gridTemplateAreas;
exports.gridTemplateColumns = gridTemplateColumns;
exports.gridTemplateRows = gridTemplateRows;
exports.height = height;
exports.justifyContent = justifyContent;
exports.justifyItems = justifyItems;
exports.justifySelf = justifySelf;
exports.left = left;
exports.lineHeight = lineHeight;
exports.margin = margin;
exports.maxHeight = maxHeight;
exports.maxWidth = maxWidth;
exports.minHeight = minHeight;
exports.minWidth = minWidth;
exports.opacity = opacity;
exports.order = order;
exports.overflow = overflow;
exports.padding = padding;
exports.position = position;
exports.positionSet = positionSet;
exports.right = right;
exports.spaceSet = spaceSet;
exports.sx = sx;
exports.textAlign = textAlign;
exports.textColor = textColor;
exports.top = top;
exports.variant = variant;
exports.width = width;
exports.zIndex = zIndex;
