// @workday/canvas-kit-react-badge/dist/commonjs/lib/CountBadge.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var core_1 = require("@emotion/core");
var styled_1 = __importDefault(require("@emotion/styled"));
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var variants = {
    default: {
        background: canvas_kit_react_core_1.colors.cinnamon500,
        color: canvas_kit_react_core_1.colors.frenchVanilla100,
        textShadow: '0 0 1px rgba(0,0,0, 0.35)',
    },
    inverse: {
        background: canvas_kit_react_core_1.colors.frenchVanilla100,
        boxShadow: '0 1px 2px rgba(0,0,0, 0.25)',
        color: canvas_kit_react_core_1.colors.blueberry400,
    },
};
var grow = core_1.keyframes(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  from {\n    transform: scale(0.85);\n  }\n\n  to {\n    transform: scale(1.0);\n  }\n"], ["\n  from {\n    transform: scale(0.85);\n  }\n\n  to {\n    transform: scale(1.0);\n  }\n"])));
var Container = styled_1.default('span')({
    alignItems: 'center',
    animation: grow + " 0.2s ease",
    borderRadius: canvas_kit_react_core_1.borderRadius.circle,
    boxSizing: 'border-box',
    display: 'inline-flex',
    fontFamily: canvas_kit_react_core_1.fontFamily,
    fontSize: '12px',
    fontWeight: 700,
    height: '20px',
    justifyContent: 'center',
    lineHeight: '20px',
    minWidth: '20px',
    padding: '0 6.5px',
}, function (_a) {
    var _b = _a.variant, variant = _b === void 0 ? 'default' : _b;
    return (__assign({}, variants[variant]));
});
var CountBadge = function (props) {
    var _a = props.variant, variant = _a === void 0 ? 'default' : _a, _b = props.count, count = _b === void 0 ? 0 : _b, elemProps = __rest(props, ["variant", "count"]);
    var formattedCount = count < 1000 ? "" + count : '999+';
    return (React.createElement(Container, __assign({ variant: variant }, elemProps), formattedCount));
};
exports.default = CountBadge;
var templateObject_1;
