// @workday/canvas-kit-react-avatar/dist/commonjs/lib/Avatar.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var canvas_kit_react_common_1 = require("@workday/canvas-kit-react-common");
var is_prop_valid_1 = __importDefault(require("@emotion/is-prop-valid"));
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var canvas_kit_react_icon_1 = require("@workday/canvas-kit-react-icon");
var canvas_system_icons_web_1 = require("@workday/canvas-system-icons-web");
var AvatarVariant;
(function (AvatarVariant) {
    AvatarVariant[AvatarVariant["Light"] = 0] = "Light";
    AvatarVariant[AvatarVariant["Dark"] = 1] = "Dark";
})(AvatarVariant = exports.AvatarVariant || (exports.AvatarVariant = {}));
var fadeTransition = 'opacity 150ms linear';
var StyledContainer = canvas_kit_react_common_1.styled('button', {
    shouldForwardProp: function (prop) { return is_prop_valid_1.default(prop) && prop !== 'size'; },
})(__assign({ background: canvas_kit_react_core_1.colors.soap200, position: 'relative', display: 'flex', alignItems: 'center', justifyContent: 'center', padding: 0, border: 0, boxSizing: 'border-box', overflow: 'hidden', borderRadius: canvas_kit_react_core_1.borderRadius.circle, '&:not([disabled])': {
        '&:focus': __assign({ outline: 'none' }, canvas_kit_react_common_1.focusRing({ separation: 2 })),
    } }, canvas_kit_react_common_1.hideMouseFocus), function (_a) {
    var size = _a.size;
    return ({
        height: size,
        width: size,
    });
}, function (_a) {
    var onClick = _a.onClick;
    return ({
        cursor: onClick ? 'pointer' : 'default',
    });
});
var StyledStack = canvas_kit_react_common_1.styled('span')({
    position: 'absolute',
    top: 0,
    left: 0,
}, function (_a) {
    var size = _a.size;
    return ({
        height: size,
        width: size,
    });
});
var StyledIcon = canvas_kit_react_common_1.styled(canvas_kit_react_icon_1.SystemIconCircle)({
    transition: fadeTransition,
}, function (_a) {
    var isImageLoaded = _a.isImageLoaded;
    return ({
        opacity: isImageLoaded ? 0 : 1,
    });
});
var StyledImage = canvas_kit_react_common_1.styled('img')({
    width: '100%',
    height: '100%',
    borderRadius: canvas_kit_react_core_1.borderRadius.circle,
    transition: fadeTransition,
}, function (_a) {
    var isLoaded = _a.isLoaded;
    return ({
        opacity: isLoaded ? 1 : 0,
    });
});
var Avatar = react_1.default.forwardRef(function (_a, ref) {
    var _b = _a.variant, variant = _b === void 0 ? AvatarVariant.Light : _b, _c = _a.size, size = _c === void 0 ? canvas_kit_react_icon_1.SystemIconCircleSize.m : _c, _d = _a.altText, altText = _d === void 0 ? 'Avatar' : _d, url = _a.url, onClick = _a.onClick, elemProps = __rest(_a, ["variant", "size", "altText", "url", "onClick"]);
    var _e = react_1.useState(false), imageLoaded = _e[0], setImageLoaded = _e[1];
    var loadImage = function () {
        if (!imageLoaded) {
            setImageLoaded(true);
        }
    };
    var background = variant === AvatarVariant.Dark ? canvas_kit_react_core_1.colors.blueberry400 : canvas_kit_react_core_1.colors.soap300;
    return (react_1.default.createElement(StyledContainer, __assign({ size: size, "aria-label": altText, onClick: onClick, disabled: onClick ? false : true, ref: ref }, elemProps),
        react_1.default.createElement(StyledStack, { size: size },
            react_1.default.createElement(StyledIcon, { icon: canvas_system_icons_web_1.userIcon, background: background, size: size, isImageLoaded: imageLoaded })),
        url && (react_1.default.createElement(StyledStack, { size: size },
            react_1.default.createElement(StyledImage, { src: url, alt: altText, onLoad: loadImage, isLoaded: imageLoaded })))));
});
Avatar.Variant = AvatarVariant;
Avatar.Size = canvas_kit_react_icon_1.SystemIconCircleSize;
exports.default = Avatar;
