// @workday/canvas-kit-react-select/dist/commonjs/lib/Select.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var canvas_kit_react_common_1 = require("@workday/canvas-kit-react-common");
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var canvas_system_icons_web_1 = require("@workday/canvas-system-icons-web");
var canvas_kit_react_icon_1 = require("@workday/canvas-kit-react-icon");
var SelectContainer = canvas_kit_react_common_1.styled('select')(__assign(__assign({}, canvas_kit_react_core_1.type.body), { border: "1px solid " + canvas_kit_react_core_1.inputColors.border, display: 'block', backgroundColor: canvas_kit_react_core_1.inputColors.background, borderRadius: canvas_kit_react_core_1.borderRadius.m, boxSizing: 'border-box', height: canvas_kit_react_core_1.spacing.xl, minWidth: 280, transition: '0.2s box-shadow, 0.2s border-color', padding: canvas_kit_react_core_1.spacingNumbers.xxs, margin: 0, MozAppearance: 'none', WebkitAppearance: 'none', '&::placeholder': {
        color: canvas_kit_react_core_1.inputColors.placeholder,
    }, '&:hover': {
        borderColor: canvas_kit_react_core_1.inputColors.hoverBorder,
    }, '&:focus:not([disabled])': {
        borderColor: canvas_kit_react_core_1.inputColors.focusBorder,
        boxShadow: "inset 0 0 0 1px " + canvas_kit_react_core_1.inputColors.focusBorder,
        outline: 'none',
    }, '&:disabled': {
        backgroundColor: canvas_kit_react_core_1.inputColors.disabled.background,
        borderColor: canvas_kit_react_core_1.inputColors.disabled.border,
        color: canvas_kit_react_core_1.inputColors.disabled.text,
        '&::placeholder': {
            color: canvas_kit_react_core_1.inputColors.disabled.text,
        },
    } }), function (_a) {
    var error = _a.error;
    return (__assign({}, canvas_kit_react_common_1.errorRing(error)));
}, function (_a) {
    var grow = _a.grow;
    return grow && {
        width: '100%',
    };
});
var SelectDropdownIcon = canvas_kit_react_common_1.styled(canvas_kit_react_icon_1.SystemIcon)({
    position: 'absolute',
    top: canvas_kit_react_core_1.spacing.xxxs,
    right: canvas_kit_react_core_1.spacing.xxxs,
    padding: canvas_kit_react_core_1.spacing.xxxs,
    pointerEvents: 'none',
    '& path': {
        transition: '100ms fill',
    },
});
var SelectWrapper = canvas_kit_react_common_1.styled('div')({
    position: 'relative',
}, function (_a) {
    var grow = _a.grow;
    return ({
        display: grow ? 'block' : 'inline-block',
    });
}, function (_a) {
    var disabled = _a.disabled;
    return ({
        '& span': {
            backgroundColor: disabled ? canvas_kit_react_core_1.colors.soap100 : canvas_kit_react_core_1.colors.frenchVanilla100,
        },
        '&:hover path': {
            fill: disabled ? undefined : canvas_kit_react_core_1.colors.licorice500,
        },
    });
});
var Select = (function (_super) {
    __extends(Select, _super);
    function Select() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Select.prototype.render = function () {
        var _a = this.props, _b = _a.disabled, disabled = _b === void 0 ? false : _b, error = _a.error, grow = _a.grow, children = _a.children, value = _a.value, onChange = _a.onChange, elemProps = __rest(_a, ["disabled", "error", "grow", "children", "value", "onChange"]);
        return (React.createElement(SelectWrapper, { grow: grow, disabled: disabled },
            React.createElement(SelectContainer, __assign({ disabled: disabled, grow: grow, error: error, value: value, onChange: onChange }, elemProps), children),
            React.createElement(SelectDropdownIcon, { icon: canvas_system_icons_web_1.caretDownSmallIcon, color: disabled ? canvas_kit_react_core_1.colors.licorice100 : canvas_kit_react_core_1.colors.licorice200, colorHover: disabled ? canvas_kit_react_core_1.colors.licorice100 : canvas_kit_react_core_1.colors.licorice500 })));
    };
    Select.ErrorType = canvas_kit_react_common_1.ErrorType;
    return Select;
}(React.Component));
exports.default = Select;
