// @storybook/design-system/dist/components/table-of-contents/BulletLink.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BulletLink = BulletLink;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styles = require("../shared/styles");

var _Link = require("../Link");

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  margin-bottom: 1px;\n  margin-right: 16px;\n  background: ", ";\n  box-shadow: white 0 0 0 4px;\n  height: 8px;\n  width: 8px;\n  border-radius: 1em;\n  text-decoration: none !important;\n  content: '';\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  && {\n    padding-top: 0;\n    list-style-type: none;\n  }\n\n  &:first-of-type ", " {\n    margin-top: -6px;\n\n    &::after {\n      height: 50%;\n      top: 50%;\n    }\n  }\n\n  &:last-of-type ", " {\n    margin-bottom: -6px;\n\n    &::after {\n      height: 50%;\n      bottom: 50%;\n    }\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  outline: none;\n  display: inline-block;\n  padding: 6px 0;\n  line-height: 1.5;\n  position: relative;\n  z-index: 1;\n  ", "\n\n  &::after {\n    position: absolute;\n    top: 0px;\n    right: auto;\n    bottom: 0px;\n    left: 3px;\n    width: auto;\n    height: auto;\n    border-left: 1px solid ", ";\n    content: '';\n    z-index: -1;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var StyledBulletLink = (0, _styledComponents["default"])(function (_ref) {
  var isActive = _ref.isActive,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["isActive"]);
  return /*#__PURE__*/_react["default"].createElement(_Link.Link, rest);
})(_templateObject(), function (props) {
  return props.isActive ? "font-weight: ".concat(_styles.typography.weight.bold, ";") : "color: ".concat(_styles.color.darker, ";");
}, _styles.color.mediumlight);

var BulletLinkWrapper = _styledComponents["default"].li(_templateObject2(), StyledBulletLink, StyledBulletLink);

var Bullet = _styledComponents["default"].span(_templateObject3(), _styles.color.medium, function (props) {
  return props.isActive && "background: ".concat(_styles.color.secondary, ";");
});

function BulletLink(_ref2) {
  var currentPath = _ref2.currentPath,
      item = _ref2.item,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["currentPath", "item"]);
  var isActive = currentPath === item.path;
  return /*#__PURE__*/_react["default"].createElement(BulletLinkWrapper, null, /*#__PURE__*/_react["default"].createElement(StyledBulletLink, {
    isActive: isActive,
    href: item.path,
    LinkWrapper: item.LinkWrapper,
    tertiary: !isActive
  }, /*#__PURE__*/_react["default"].createElement(Bullet, {
    isActive: isActive
  }), item.title));
}

BulletLink.propTypes = {
  currentPath: _propTypes["default"].string.isRequired,
  item: _propTypes["default"].shape({
    LinkWrapper: _propTypes["default"].oneOfType([_propTypes["default"].func, _propTypes["default"].object]),
    path: _propTypes["default"].string.isRequired,
    title: _propTypes["default"].string.isRequired
  }).isRequired
};
