// @storybook/design-system/dist/components/table-of-contents/ItemLink.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ItemLink = ItemLink;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styles = require("../shared/styles");

var _Link = require("../Link");

var _MenuLink = require("./MenuLink");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  list-style-type: none;\n\n  &:last-of-type ", " {\n    margin-bottom: 0;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var ItemLinkWrapper = _styledComponents["default"].li(_templateObject(), _MenuLink.MenuLink);

function ItemLink(_ref) {
  var currentPath = _ref.currentPath,
      item = _ref.item;
  return /*#__PURE__*/_react["default"].createElement(ItemLinkWrapper, null, /*#__PURE__*/_react["default"].createElement(_MenuLink.MenuLink, {
    isActive: currentPath === item.path,
    href: item.path,
    LinkWrapper: item.LinkWrapper,
    tertiary: true
  }, item.title));
}

ItemLink.propTypes = {
  currentPath: _propTypes["default"].string.isRequired,
  item: _propTypes["default"].shape({
    path: _propTypes["default"].string.isRequired,
    LinkWrapper: _propTypes["default"].oneOfType([_propTypes["default"].func, _propTypes["default"].object]),
    title: _propTypes["default"].string.isRequired
  }).isRequired
};
