// @storybook/design-system/dist/components/table-of-contents/TableOfContentsItems.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableOfContentsItems = TableOfContentsItems;
exports.ITEM_TYPES = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _BulletLink = require("./BulletLink");

var _ItemLink = require("./ItemLink");

var _MenuLink = require("./MenuLink");

var _styles = require("../shared/styles");

var _Icon = require("../Icon");

var _Link = require("../Link");

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  list-style-type: none;\n  padding: 0;\n  margin: 0;\n\n  li {\n    padding-top: 12px;\n\n    &:first-child {\n      padding-top: 0;\n    }\n  }\n\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  && {\n    width: 12px;\n    height: 12px;\n    color: ", ";\n    transform: translateY(1px) ", ";\n    ", "\n    bottom: -0.25em;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-weight: ", ";\n  line-height: 24px;\n  word-break: break-word;\n  text-align: left;\n\n  > span {\n    display: flex;\n\n    svg {\n      flex: none;\n    }\n\n    span {\n      flex: 1;\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var MENU = 'menu';
var LINK = 'link';
var BULLET_LINK = 'bullet-link';
var ITEM_TYPES = Object.freeze({
  MENU: MENU,
  LINK: LINK,
  BULLET_LINK: BULLET_LINK
});
exports.ITEM_TYPES = ITEM_TYPES;

var getItemComponent = function getItemComponent(itemType) {
  switch (itemType) {
    case ITEM_TYPES.MENU:
      return Menu;

    case ITEM_TYPES.LINK:
      return _ItemLink.ItemLink;

    case ITEM_TYPES.BULLET_LINK:
      return _BulletLink.BulletLink;

    default:
      return null;
  }
};

var TopLevelMenuToggle = (0, _styledComponents["default"])(_Link.Link).attrs({
  isButton: true,
  tertiary: true
})(_templateObject(), _styles.typography.weight.bold);
var ArrowIcon = (0, _styledComponents["default"])(_Icon.Icon).attrs({
  icon: 'arrowright'
})(_templateObject2(), _styles.color.mediumdark, function (props) {
  return props.isOpen && "rotate(90deg)";
}, function (props) {
  return props.isTopLevel ? "margin-right: 8px;" : "margin-left: 8px;";
});

function Menu(_ref) {
  var isTopLevel = _ref.isTopLevel,
      item = _ref.item,
      setMenuOpenStateById = _ref.setMenuOpenStateById,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["isTopLevel", "item", "setMenuOpenStateById"]);
  if (!item.children) return null;

  var arrow = /*#__PURE__*/_react["default"].createElement(ArrowIcon, {
    isOpen: item.isOpen,
    isTopLevel: isTopLevel,
    "aria-hidden": true
  });

  var MenuToggle = isTopLevel ? TopLevelMenuToggle : _MenuLink.MenuLink;

  var toggleOpenState = function toggleOpenState() {
    return setMenuOpenStateById({
      id: item.id,
      isOpen: !item.isOpen
    });
  };

  return /*#__PURE__*/_react["default"].createElement("li", null, isTopLevel ? /*#__PURE__*/_react["default"].createElement(TopLevelMenuToggle, {
    onClick: toggleOpenState
  }, arrow, /*#__PURE__*/_react["default"].createElement("span", null, item.title)) : /*#__PURE__*/_react["default"].createElement(_MenuLink.MenuLink, {
    isButton: true,
    onClick: toggleOpenState
  }, item.title, arrow), item.isOpen && /*#__PURE__*/_react["default"].createElement(TableOfContentsItems, (0, _extends2["default"])({
    items: item.children,
    isTopLevel: false,
    setMenuOpenStateById: setMenuOpenStateById
  }, rest)));
}

Menu.propTypes = {
  isTopLevel: _propTypes["default"].bool,
  item: _propTypes["default"].shape({
    id: _propTypes["default"].string.isRequired,
    isOpen: _propTypes["default"].bool.isRequired,
    title: _propTypes["default"].string.isRequired,
    children: _propTypes["default"].arrayOf(_propTypes["default"].shape({}))
  }).isRequired,
  setMenuOpenStateById: _propTypes["default"].func.isRequired
};
Menu.defaultProps = {
  isTopLevel: false
};

var List = _styledComponents["default"].ul(_templateObject3(), function (props) {
  return props.isTopLevel && "  \n    > li {\n      padding-top: 24px;\n\n      ul, ol {\n        padding-top: 12px;\n        display: flex;\n        flex-direction: column;\n      }\n      \n      > ul { \n        padding-left: 35px;\n\n        > li ul {\n          padding-left: 15px;\n        }\n      }\n\n      > ol {\n        padding-left: 20px;\n    \n      }\n    }\n  ";
});

function TableOfContentsItems(_ref2) {
  var className = _ref2.className,
      currentPath = _ref2.currentPath,
      isTopLevel = _ref2.isTopLevel,
      items = _ref2.items,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["className", "currentPath", "isTopLevel", "items"]);
  var isOrderedList = items.every(function (item) {
    return item.type === ITEM_TYPES.BULLET_LINK;
  });
  return /*#__PURE__*/_react["default"].createElement(List, {
    className: className,
    isTopLevel: isTopLevel,
    as: isOrderedList ? 'ol' : 'ul'
  }, items.map(function (item) {
    var ItemComponent = getItemComponent(item.type);
    return /*#__PURE__*/_react["default"].createElement(ItemComponent, (0, _extends2["default"])({
      key: item.title,
      currentPath: currentPath,
      item: item,
      isTopLevel: isTopLevel
    }, rest));
  }));
}

TableOfContentsItems.propTypes = {
  className: _propTypes["default"].string,
  currentPath: _propTypes["default"].string.isRequired,
  isTopLevel: _propTypes["default"].bool.isRequired,
  items: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    type: _propTypes["default"].oneOf(Object.values(ITEM_TYPES)).isRequired
  }).isRequired).isRequired
};
TableOfContentsItems.defaultProps = {
  className: ''
};
