// @workday/canvas-kit-react-tooltip/dist/commonjs/lib/TooltipContainer.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@emotion/core");
var styled_1 = __importDefault(require("@emotion/styled"));
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var canvas_kit_react_common_1 = require("@workday/canvas-kit-react-common");
var tooltipAnimation = function (transformOrigin) {
    var translate = canvas_kit_react_common_1.getTranslateFromOrigin(transformOrigin, canvas_kit_react_core_1.spacing.xxxs);
    return core_1.keyframes(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    0% {\n      opacity: 0;\n      transform: translate(", "px, ", "px);\n    }\n    100% {\n      opacity: 1;\n      transform: translate(0);\n    }\n  "], ["\n    0% {\n      opacity: 0;\n      transform: translate(", "px, ", "px);\n    }\n    100% {\n      opacity: 1;\n      transform: translate(0);\n    }\n  "])), translate.x, translate.y);
};
var defaultTransformOrigin = {
    vertical: 'bottom',
    horizontal: 'center',
};
exports.TooltipContainer = styled_1.default('div')(__assign(__assign({}, canvas_kit_react_core_1.type.body), { display: 'inline-flex', padding: canvas_kit_react_core_1.spacingNumbers.xxs + canvas_kit_react_core_1.spacingNumbers.xxxs, color: canvas_kit_react_core_1.colors.frenchVanilla100, fontSize: 13, a: {
        color: canvas_kit_react_core_1.colors.frenchVanilla100,
        textDecoration: 'underline',
    }, ':before': {
        content: '""',
        borderRadius: canvas_kit_react_core_1.borderRadius.m,
        zIndex: -1,
        margin: canvas_kit_react_core_1.spacingNumbers.xxxs,
        backgroundColor: 'rgba(0,0,0,.85)',
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
    } }), function (_a) {
    var _b = _a.transformOrigin, transformOrigin = _b === void 0 ? defaultTransformOrigin : _b;
    if (transformOrigin === null) {
        return {};
    }
    return {
        animation: tooltipAnimation(transformOrigin),
        animationDuration: '150ms',
        animationTimingFunction: 'ease-out',
        transformOrigin: transformOrigin
            ? transformOrigin.vertical + " " + transformOrigin.horizontal
            : 'top center',
    };
});
var templateObject_1;
