// @workday/canvas-kit-react-segmented-control/dist/es6/lib/SegmentedControl.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { borderRadius, colors } from '@workday/canvas-kit-react-core';
import { IconButton } from '@workday/canvas-kit-react-button';
import { mouseFocusBehavior, styled } from '@workday/canvas-kit-react-common';
var SegmentedControlContainer = styled('div')({
    '& button': {
        borderRadius: borderRadius.zero,
        border: "1px solid " + colors.soap500,
        borderLeft: 'none',
        '&[aria-pressed="true"]': {
            borderColor: colors.blueberry400,
            '&:hover, &:focus:hover': {
                background: colors.blueberry400,
            },
        },
        '&:first-of-type': {
            borderRadius: borderRadius.m + " 0 0 " + borderRadius.m,
            borderLeft: "1px solid " + colors.soap500,
        },
        '&:last-of-type': {
            borderRadius: "0 " + borderRadius.m + " " + borderRadius.m + " 0",
        },
        '&:focus': __assign({ borderRadius: borderRadius.m, zIndex: 1, animation: 'none', transition: 'all 120ms, border-radius 1ms' }, mouseFocusBehavior({
            '&': {
                borderRadius: borderRadius.zero,
                '&:first-of-type': {
                    borderRadius: borderRadius.m + " 0 0 " + borderRadius.m,
                },
                '&:last-of-type': {
                    borderRadius: "0 " + borderRadius.m + " " + borderRadius.m + " 0",
                },
            },
        })),
    },
}, function (_a) {
    var theme = _a.theme;
    return ({
        '& button[aria-pressed="true"]': {
            borderColor: theme.canvas.palette.primary.main,
            '&:hover, &:focus:hover': {
                background: theme.canvas.palette.primary.main,
            },
        },
    });
});
var onButtonClick = function (existingOnClick, onChange, index, event) {
    if (existingOnClick) {
        existingOnClick(event);
    }
    var target = event.currentTarget;
    if (target && onChange) {
        if (target.value) {
            onChange(target.value);
        }
        else {
            onChange(index);
        }
    }
};
var SegmentedControl = function (_a) {
    var _b = _a.value, value = _b === void 0 ? 0 : _b, children = _a.children, onChange = _a.onChange, elemProps = __rest(_a, ["value", "children", "onChange"]);
    return (React.createElement(SegmentedControlContainer, __assign({}, elemProps), React.Children.map(children, function (child, index) {
        if (typeof child.type === typeof IconButton) {
            return React.cloneElement(child, {
                toggled: typeof value === 'number' ? index === value : child.props.value === value,
                variant: IconButton.Variant.SquareFilled,
                onClick: onButtonClick.bind(undefined, child.props.onClick, onChange, index),
            });
        }
        return child;
    })));
};
export default SegmentedControl;
