// @workday/canvas-kit-react-table/dist/commonjs/lib/TableRow.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var styled_1 = __importDefault(require("@emotion/styled"));
var polished_1 = require("polished");
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var Table_1 = require("./Table");
var TableRowState;
(function (TableRowState) {
    TableRowState[TableRowState["Error"] = 0] = "Error";
    TableRowState[TableRowState["Alert"] = 1] = "Alert";
    TableRowState[TableRowState["InputError"] = 2] = "InputError";
    TableRowState[TableRowState["InputAlert"] = 3] = "InputAlert";
    TableRowState[TableRowState["Hover"] = 4] = "Hover";
    TableRowState[TableRowState["Selected"] = 5] = "Selected";
})(TableRowState = exports.TableRowState || (exports.TableRowState = {}));
var errorColor = canvas_kit_react_core_1.statusColors.error;
var errorColorLight = canvas_kit_react_core_1.colors.cinnamon200;
var alertColor = canvas_kit_react_core_1.statusColors.warning;
var alertColorLight = canvas_kit_react_core_1.colors.cantaloupe200;
function makeColoredRow(_bgColor, _borderColor) {
    return {
        backgroundColor: _bgColor,
        position: 'relative',
        '&:before, &:after': {
            position: 'absolute',
            left: 0,
            top: -Table_1.borderWidth,
            content: '""',
            width: "calc(100% + " + Table_1.borderWidth + "px)",
            height: Table_1.borderWidth,
            backgroundColor: _borderColor,
        },
        '&:after': {
            top: 'auto',
            bottom: -Table_1.borderWidth,
        },
        '&:first-of-type': {
            boxShadow: "inset " + Table_1.borderWidth + "px 0 0 " + _borderColor,
        },
        '&:last-child': {
            boxShadow: "inset -" + Table_1.borderWidth + "px 0 0 " + _borderColor,
            '&:before, &:after': {
                width: '100%',
            },
        },
    };
}
function makeColoredRowStyle(_bgColor, _borderColor) {
    var lightenedBg = polished_1.rgba(_bgColor, 0.2);
    return {
        'th, td': [
            makeColoredRow(lightenedBg, _bgColor),
            {
                borderLeftColor: _borderColor,
                borderRightColor: _borderColor,
            },
        ],
        '&:hover': {
            'th, td': {
                backgroundColor: lightenedBg,
            },
        },
    };
}
function makeBorderlessStyle(_bgColor) {
    return {
        'th, td': {
            backgroundColor: polished_1.rgba(_bgColor, 0.2),
        },
    };
}
var Row = styled_1.default('tr')({
    'th, td': {
        backgroundColor: canvas_kit_react_core_1.colors.frenchVanilla100,
        padding: canvas_kit_react_core_1.spacing.xs + " " + canvas_kit_react_core_1.spacing.xxs,
        fontSize: 13,
        borderRight: Table_1.cellBorder,
        borderBottom: Table_1.cellBorder,
        height: canvas_kit_react_core_1.spacingNumbers.xl + canvas_kit_react_core_1.spacingNumbers.xxs,
        boxSizing: 'border-box',
        transition: 'background-color 0.2s',
    },
    td: {
        '&:last-child': {
            borderRight: 'none',
        },
    },
    th: {
        height: canvas_kit_react_core_1.spacing.xxl,
        '&:last-child': {
            borderRight: 'none',
        },
    },
    '&:last-child': {
        'th, td': {
            borderBottom: 'none',
        },
    },
}, function (_a) {
    var header = _a.header;
    return header && {
        '&:last-child': {
            'th, td': {
                borderBottom: '1px solid transparent',
            },
        },
        'th, td': {
            position: 'relative',
            backgroundColor: canvas_kit_react_core_1.colors.frenchVanilla100,
            verticalAlign: 'middle',
            '&:after': {
                position: 'absolute',
                left: 0,
                bottom: -Table_1.borderWidth,
                content: '""',
                width: "calc(100% + " + Table_1.borderWidth + "px)",
                height: Table_1.borderWidth,
                backgroundColor: Table_1.borderColor,
            },
            '&:last-child': {
                '&:after': {
                    width: '100%',
                },
            },
        },
    };
}, function (_a) {
    var state = _a.state;
    switch (state) {
        case TableRowState.InputError:
            return makeBorderlessStyle(errorColor);
        case TableRowState.Error:
            return makeColoredRowStyle(errorColor, errorColorLight);
        case TableRowState.InputAlert:
            return makeBorderlessStyle(alertColor);
        case TableRowState.Alert:
            return makeColoredRowStyle(alertColor, alertColorLight);
        case TableRowState.Selected:
            return {
                'th, td': [
                    makeColoredRow(canvas_kit_react_core_1.colors.blueberry100, canvas_kit_react_core_1.colors.blueberry500),
                    {
                        '&:after': {
                            zIndex: 2,
                        },
                    },
                ],
            };
        default:
            return {
                '&:hover': {
                    'th, td': {
                        backgroundColor: canvas_kit_react_core_1.colors.soap300,
                    },
                },
                '&:focus': {
                    outline: 'none',
                    'th, td': [
                        makeColoredRow(canvas_kit_react_core_1.colors.blueberry100, canvas_kit_react_core_1.colors.blueberry500),
                        {
                            '&:after': {
                                zIndex: 2,
                            },
                        },
                    ],
                },
            };
    }
});
var TableRow = (function (_super) {
    __extends(TableRow, _super);
    function TableRow() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    TableRow.prototype.render = function () {
        var _a = this.props, state = _a.state, header = _a.header, children = _a.children, elemProps = __rest(_a, ["state", "header", "children"]);
        return (React.createElement(Row, __assign({ state: state, header: header }, elemProps), children));
    };
    TableRow.State = TableRowState;
    return TableRow;
}(React.Component));
exports.default = TableRow;
