// @workday/canvas-kit-react-tooltip/dist/es6/lib/TooltipContainer.js
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { keyframes } from '@emotion/core';
import styled from '@emotion/styled';
import { borderRadius, colors, spacing, spacingNumbers, type } from '@workday/canvas-kit-react-core';
import { getTranslateFromOrigin } from '@workday/canvas-kit-react-common';
var tooltipAnimation = function (transformOrigin) {
    var translate = getTranslateFromOrigin(transformOrigin, spacing.xxxs);
    return keyframes(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    0% {\n      opacity: 0;\n      transform: translate(", "px, ", "px);\n    }\n    100% {\n      opacity: 1;\n      transform: translate(0);\n    }\n  "], ["\n    0% {\n      opacity: 0;\n      transform: translate(", "px, ", "px);\n    }\n    100% {\n      opacity: 1;\n      transform: translate(0);\n    }\n  "])), translate.x, translate.y);
};
var defaultTransformOrigin = {
    vertical: 'bottom',
    horizontal: 'center',
};
export var TooltipContainer = styled('div')(__assign(__assign({}, type.body), { display: 'inline-flex', padding: spacingNumbers.xxs + spacingNumbers.xxxs, color: colors.frenchVanilla100, fontSize: 13, a: {
        color: colors.frenchVanilla100,
        textDecoration: 'underline',
    }, ':before': {
        content: '""',
        borderRadius: borderRadius.m,
        zIndex: -1,
        margin: spacingNumbers.xxxs,
        backgroundColor: 'rgba(0,0,0,.85)',
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
    } }), function (_a) {
    var _b = _a.transformOrigin, transformOrigin = _b === void 0 ? defaultTransformOrigin : _b;
    if (transformOrigin === null) {
        return {};
    }
    return {
        animation: tooltipAnimation(transformOrigin),
        animationDuration: '150ms',
        animationTimingFunction: 'ease-out',
        transformOrigin: transformOrigin
            ? transformOrigin.vertical + " " + transformOrigin.horizontal
            : 'top center',
    };
});
var templateObject_1;
