// @workday/canvas-kit-react-switch/dist/es6/lib/Switch.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { ErrorType, focusRing, mouseFocusBehavior, getErrorColors, styled, useUniqueId, } from '@workday/canvas-kit-react-common';
import { borderRadius, colors, depth, spacing } from '@workday/canvas-kit-react-core';
var circleSize = spacing.xs;
var switchWidth = spacing.l;
var switchHeight = spacing.s;
var switchTapArea = spacing.l;
var translateLength = spacing.s;
var SwitchContainer = styled('div')({
    position: 'relative',
    height: spacing.m,
    width: switchTapArea,
});
var SwitchInput = styled('input')({
    position: 'absolute',
    height: spacing.m,
    width: switchTapArea,
    margin: 0,
    marginLeft: spacing.xxxs,
    borderRadius: borderRadius.circle,
    opacity: 0,
    display: 'block',
}, function (_a) {
    var disabled = _a.disabled;
    return ({
        cursor: disabled ? 'not-allowed' : 'pointer',
    });
}, function (_a) {
    var error = _a.error, theme = _a.theme;
    var errorColors = getErrorColors(error, theme);
    if (errorColors.outer === errorColors.inner) {
        errorColors.outer = 'transparent';
    }
    var styles = {
        '&:focus': {
            outline: 'none',
            '& ~ div:first-of-type': __assign({}, focusRing({ separation: 2, animate: false }, theme)),
        },
        '& ~ div:first-of-type': {
            boxShadow: "\n          0 0 0 2px " + colors.frenchVanilla100 + ",\n          0 0 0 4px " + errorColors.inner + ",\n          0 0 0 5px " + errorColors.outer,
        },
        '&:focus ~ div:first-of-type': __assign({}, focusRing({ separation: 2, animate: false }, theme)),
    };
    return __assign(__assign({}, styles), mouseFocusBehavior(__assign(__assign({}, styles), { '&:focus ~ div:first-of-type, &:active ~ div:first-of-type': {
            boxShadow: "\n            0 0 0 2px " + colors.frenchVanilla100 + ",\n            0 0 0 4px " + errorColors.inner + ",\n            0 0 0 5px " + errorColors.outer,
        } })));
});
var SwitchBackground = styled('div')({
    boxSizing: 'border-box',
    position: 'absolute',
    display: 'flex',
    alignItems: 'center',
    pointerEvents: 'none',
    marginTop: spacing.xxxs,
    width: switchWidth,
    height: switchHeight,
    borderRadius: borderRadius.circle,
    padding: '0px 2px',
    transition: 'background-color 200ms ease',
}, function (_a) {
    var checked = _a.checked, disabled = _a.disabled, themePrimary = _a.theme.canvas.palette.primary;
    if (checked) {
        return {
            backgroundColor: disabled ? themePrimary.light : themePrimary.main,
        };
    }
    else {
        return {
            backgroundColor: disabled ? colors.soap400 : colors.licorice200,
        };
    }
});
var SwitchCircle = styled('div')(__assign(__assign({ width: circleSize, height: circleSize, borderRadius: borderRadius.circle }, depth[1]), { transition: 'transform 150ms ease', pointerEvents: 'none' }), function (_a) {
    var theme = _a.theme;
    return ({
        backgroundColor: theme.canvas.palette.primary.contrast,
    });
}, function (_a) {
    var checked = _a.checked;
    return ({
        transform: checked ? "translateX(" + translateLength + ")" : 'translateX(0)',
    });
});
export var Switch = function (_a) {
    var _b = _a.checked, checked = _b === void 0 ? false : _b, id = _a.id, _c = _a.disabled, disabled = _c === void 0 ? false : _c, inputRef = _a.inputRef, onChange = _a.onChange, value = _a.value, elemProps = __rest(_a, ["checked", "id", "disabled", "inputRef", "onChange", "value"]);
    var inputId = useUniqueId(id);
    return (React.createElement(SwitchContainer, null,
        React.createElement(SwitchInput, __assign({ checked: checked, disabled: disabled, id: inputId, ref: inputRef, onChange: onChange, role: "checkbox", tabIndex: 0, type: "checkbox", value: value }, elemProps)),
        React.createElement(SwitchBackground, { checked: checked, disabled: disabled },
            React.createElement(SwitchCircle, { checked: checked }))));
};
Switch.ErrorType = ErrorType;
export default Switch;
