// @workday/canvas-kit-react-page-header/dist/es6/lib/PageHeader.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import styled from '@emotion/styled';
import { colors, gradients, spacing, type } from '@workday/canvas-kit-react-core';
import { IconButton } from '@workday/canvas-kit-react-button';
var Header = styled('header')({
    height: 84,
    backgroundImage: gradients.blueberry,
    color: colors.frenchVanilla100,
    WebkitFontSmoothing: 'antialiased',
    MozOsxFontSmoothing: 'grayscale',
});
var Container = styled('div')({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    height: '100%',
    overflow: 'hidden',
    padding: "0 " + spacing.s,
}, function (_a) {
    var _b;
    var breakpoint = _a.breakpoint;
    return (_b = {},
        _b["@media (min-width: " + breakpoint + "px)"] = {
            padding: "0 " + spacing.xl,
        },
        _b);
}, function (_a) {
    var capWidth = _a.capWidth;
    return capWidth && {
        boxSizing: 'border-box',
        margin: '0 auto',
        width: '100%',
        maxWidth: 1440,
    };
});
var Title = styled('h2')(__assign(__assign({}, type.h1), { color: colors.frenchVanilla100, padding: spacing.xs + " 0", margin: 0, whiteSpace: 'nowrap' }));
var IconList = styled('div')({
    display: 'flex',
    marginLeft: spacing.l,
    '> *:not(:last-child)': {
        marginRight: spacing.xxs,
    },
});
var PageHeader = (function (_super) {
    __extends(PageHeader, _super);
    function PageHeader() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    PageHeader.prototype.renderChildren = function (children) {
        return React.Children.map(children, function (child) {
            if (!React.isValidElement(child)) {
                return child;
            }
            if (child.type === IconButton) {
                return React.cloneElement(child, {
                    variant: IconButton.Variant.Inverse,
                });
            }
            return child;
        });
    };
    PageHeader.prototype.render = function () {
        var _a = this.props, _b = _a.title, title = _b === void 0 ? '' : _b, _c = _a.breakpoint, breakpoint = _c === void 0 ? 768 : _c, _d = _a.capWidth, capWidth = _d === void 0 ? false : _d, children = _a.children, elemProps = __rest(_a, ["title", "breakpoint", "capWidth", "children"]);
        return (React.createElement(Header, null,
            React.createElement(Container, __assign({ breakpoint: breakpoint, capWidth: capWidth }, elemProps),
                React.createElement(Title, null, title),
                React.createElement(IconList, null, this.renderChildren(children)))));
    };
    return PageHeader;
}(React.Component));
export default PageHeader;
