// @workday/canvas-kit-react-status-indicator/dist/commonjs/lib/StatusIndicator.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var canvas_kit_react_icon_1 = require("@workday/canvas-kit-react-icon");
var canvas_kit_react_core_1 = require("@workday/canvas-kit-react-core");
var styled_1 = __importDefault(require("@emotion/styled"));
var StatusIndicatorType;
(function (StatusIndicatorType) {
    StatusIndicatorType["Gray"] = "gray";
    StatusIndicatorType["Orange"] = "orange";
    StatusIndicatorType["Blue"] = "blue";
    StatusIndicatorType["Green"] = "green";
    StatusIndicatorType["Red"] = "red";
    StatusIndicatorType["Transparent"] = "transparent";
})(StatusIndicatorType = exports.StatusIndicatorType || (exports.StatusIndicatorType = {}));
var StatusIndicatorEmphasis;
(function (StatusIndicatorEmphasis) {
    StatusIndicatorEmphasis["High"] = "high";
    StatusIndicatorEmphasis["Low"] = "low";
})(StatusIndicatorEmphasis = exports.StatusIndicatorEmphasis || (exports.StatusIndicatorEmphasis = {}));
exports.statusIndicatorStyles = {
    classname: 'status-indicator',
    styles: __assign(__assign({}, canvas_kit_react_core_1.type.small), { display: 'inline-flex', alignItems: 'center', verticalAlign: 'middle', maxWidth: 150, height: canvas_kit_react_core_1.spacing.s, padding: "1px " + canvas_kit_react_core_1.spacing.xxxs, borderRadius: canvas_kit_react_core_1.borderRadius.s, boxSizing: 'border-box' }),
    variants: {
        gray: {
            high: {
                color: canvas_kit_react_core_1.colors.frenchVanilla100,
                background: canvas_kit_react_core_1.colors.licorice300,
            },
            low: {
                color: canvas_kit_react_core_1.colors.licorice400,
                background: canvas_kit_react_core_1.colors.soap200,
            },
        },
        orange: {
            high: {
                color: canvas_kit_react_core_1.colors.licorice500,
                background: canvas_kit_react_core_1.colors.cantaloupe400,
            },
            low: {
                color: canvas_kit_react_core_1.colors.toastedMarshmallow600,
                background: canvas_kit_react_core_1.colors.cantaloupe100,
            },
        },
        blue: {
            high: {
                color: canvas_kit_react_core_1.colors.frenchVanilla100,
                background: canvas_kit_react_core_1.colors.blueberry400,
            },
            low: {
                color: canvas_kit_react_core_1.colors.blueberry500,
                background: canvas_kit_react_core_1.colors.blueberry100,
            },
        },
        green: {
            high: {
                color: canvas_kit_react_core_1.colors.frenchVanilla100,
                background: canvas_kit_react_core_1.colors.greenApple600,
            },
            low: {
                color: canvas_kit_react_core_1.colors.greenApple600,
                background: canvas_kit_react_core_1.colors.greenApple100,
            },
        },
        red: {
            high: {
                color: canvas_kit_react_core_1.colors.frenchVanilla100,
                background: canvas_kit_react_core_1.colors.cinnamon500,
            },
            low: {
                color: canvas_kit_react_core_1.colors.cinnamon600,
                background: canvas_kit_react_core_1.colors.cinnamon100,
            },
        },
        transparent: {
            high: {
                color: canvas_kit_react_core_1.colors.frenchVanilla100,
                background: canvas_kit_react_core_1.colors.blackPepper600,
            },
            low: {
                color: canvas_kit_react_core_1.colors.frenchVanilla100,
                background: canvas_kit_react_core_1.colors.blackPepper600,
            },
        },
    },
};
var Container = styled_1.default('span')(exports.statusIndicatorStyles.styles, function (_a) {
    var type = _a.type, emphasis = _a.emphasis;
    return (__assign({}, exports.statusIndicatorStyles.variants[type][emphasis]));
});
var StatusLabel = styled_1.default('span')({
    fontWeight: 'bold',
    WebkitFontSmoothing: 'antialiased',
    textTransform: 'uppercase',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
});
var StatusIndicator = (function (_super) {
    __extends(StatusIndicator, _super);
    function StatusIndicator() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StatusIndicator.prototype.render = function () {
        var _a = this.props, _b = _a.emphasis, emphasis = _b === void 0 ? StatusIndicatorEmphasis.High : _b, type = _a.type, icon = _a.icon, label = _a.label, elemProps = __rest(_a, ["emphasis", "type", "icon", "label"]);
        var variant = exports.statusIndicatorStyles.variants[type][emphasis];
        return (React.createElement(Container, __assign({ type: type, emphasis: emphasis }, elemProps),
            icon && (React.createElement(canvas_kit_react_icon_1.SystemIcon, { colorHover: variant.color, color: variant.color, icon: icon, size: 14, style: { paddingRight: canvas_kit_react_core_1.spacing.xxxs } })),
            React.createElement(StatusLabel, null, label)));
    };
    StatusIndicator.Type = StatusIndicatorType;
    StatusIndicator.Emphasis = StatusIndicatorEmphasis;
    return StatusIndicator;
}(React.Component));
exports.default = StatusIndicator;
