// @storybook/design-system/dist/components/Avatar.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Avatar = exports.sizes = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styles = require("./shared/styles");

var _animation = require("./shared/animation");

var _Icon = require("./Icon");

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    font-size: ", "px;\n    line-height: ", "px;\n  "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    font-size: ", "px;\n    line-height: ", "px;\n  "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    font-size: ", "px;\n    line-height: ", "px;\n  "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  color: ", ";\n  text-align: center;\n\n  font-size: ", "px;\n  line-height: ", "px;\n\n  ", "\n\n  ", "\n\n  ", "\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      background: ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      height: ", "px;\n      width: ", "px;\n      line-height: ", "px;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      height: ", "px;\n      width: ", "px;\n      line-height: ", "px;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      height: ", "px;\n      width: ", "px;\n      line-height: ", "px;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  background: ", ";\n  border-radius: 50%;\n  display: inline-block;\n  vertical-align: top;\n  overflow: hidden;\n  text-transform: uppercase;\n\n  height: ", "px;\n  width: ", "px;\n  line-height: ", "px;\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  img {\n    width: 100%;\n    height: auto;\n    display: block;\n  }\n\n  svg {\n    position: relative;\n    bottom: -2px;\n    height: 100%;\n    width: 100%;\n    vertical-align: top;\n  }\n\n  path {\n    fill: ", ";\n    animation: ", " 1.5s ease-in-out infinite;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var sizes = {
  large: 40,
  medium: 28,
  small: 20,
  tiny: 16
};
exports.sizes = sizes;

var Image = _styledComponents["default"].div(_templateObject(), function (props) {
  return !props.isLoading ? 'transparent' : _styles.color.light;
}, sizes.medium, sizes.medium, sizes.medium, function (props) {
  return props.size === 'tiny' && (0, _styledComponents.css)(_templateObject2(), sizes.tiny, sizes.tiny, sizes.tiny);
}, function (props) {
  return props.size === 'small' && (0, _styledComponents.css)(_templateObject3(), sizes.small, sizes.small, sizes.small);
}, function (props) {
  return props.size === 'large' && (0, _styledComponents.css)(_templateObject4(), sizes.large, sizes.large, sizes.large);
}, function (props) {
  return !props.src && (0, _styledComponents.css)(_templateObject5(), !props.isLoading && '#37D5D3');
}, _styles.color.medium, _animation.glow); // prettier-ignore


var Initial = _styledComponents["default"].div(_templateObject6(), _styles.color.lightest, _styles.typography.size.s2, sizes.medium, function (props) {
  return props.size === "tiny" && (0, _styledComponents.css)(_templateObject7(), _styles.typography.size.s1 - 2, sizes.tiny);
}, function (props) {
  return props.size === "small" && (0, _styledComponents.css)(_templateObject8(), _styles.typography.size.s1, sizes.small);
}, function (props) {
  return props.size === "large" && (0, _styledComponents.css)(_templateObject9(), _styles.typography.size.s3, sizes.large);
});
/**
 * The `Avatar` component is where all your avatars come to play.
 */


var Avatar = function Avatar(_ref) {
  var _ref$isLoading = _ref.isLoading,
      isLoading = _ref$isLoading === void 0 ? false : _ref$isLoading,
      _ref$username = _ref.username,
      username = _ref$username === void 0 ? 'loading' : _ref$username,
      _ref$src = _ref.src,
      src = _ref$src === void 0 ? null : _ref$src,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'medium' : _ref$size,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["isLoading", "username", "src", "size"]);

  var avatarFigure = /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "useralt"
  });

  var a11yProps = {};

  if (isLoading) {
    a11yProps['aria-busy'] = true;
    a11yProps['aria-label'] = 'Loading avatar ...';
  } else if (src) {
    avatarFigure = /*#__PURE__*/_react["default"].createElement("img", {
      src: src,
      alt: username
    });
  } else {
    a11yProps['aria-label'] = username;
    avatarFigure = /*#__PURE__*/_react["default"].createElement(Initial, {
      size: size,
      "aria-hidden": "true"
    }, username.substring(0, 1));
  }

  return /*#__PURE__*/_react["default"].createElement(Image, (0, _extends2["default"])({
    size: size,
    isLoading: isLoading,
    src: src
  }, a11yProps, props), avatarFigure);
};

exports.Avatar = Avatar;
