// @storybook/design-system/dist/components/Button.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Button = exports.StyledButton = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _polished = require("polished");

var _styles = require("./shared/styles");

var _animation = require("./shared/animation");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  border: 0;\n  border-radius: 3em;\n  cursor: pointer;\n  display: inline-block;\n  overflow: hidden;\n  padding: ", ";\n  position: relative;\n  text-align: center;\n  text-decoration: none;\n  transition: all 150ms ease-out;\n  transform: translate3d(0,0,0);\n  vertical-align: top;\n  white-space: nowrap;\n  user-select: none;\n  opacity: 1;\n  margin: 0;\n  background: transparent;\n\n  font-size: ", "px;\n  font-weight: ", ";\n  line-height: 1;\n\n  ", "\n\n  ", " {\n    transform: scale3d(1,1,1) translate3d(0,0,0);\n    transition: transform 700ms ", ";\n    opacity: 1;\n  }\n\n  ", " {\n    transform: translate3d(0, 100%, 0);\n  }\n\n  svg {\n    height: ", "px;\n    width: ", "px;\n    vertical-align: top;\n    margin-right: ", "px;\n    margin-top: ", "px;\n    margin-bottom: ", "px;\n\n    /* Necessary for js mouse events to not glitch out when hovering on svgs */\n    pointer-events: none;\n  }\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n    ", "\n\n    ", "\n\n    ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  top: 50%;\n  left: 0;\n  right: 0;\n  opacity: 0;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  vertical-align: top;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Text = _styledComponents["default"].span(_templateObject());

var Loading = _styledComponents["default"].span(_templateObject2());

var APPEARANCES = {
  PRIMARY: 'primary',
  PRIMARY_OUTLINE: 'primaryOutline',
  SECONDARY: 'secondary',
  SECONDARY_OUTLINE: 'secondaryOutline',
  TERTIARY: 'tertiary',
  OUTLINE: 'outline',
  INVERSE_PRIMARY: 'inversePrimary',
  INVERSE_SECONDARY: 'inverseSecondary',
  INVERSE_OUTLINE: 'inverseOutline'
};
var SIZES = {
  SMALL: 'small',
  MEDIUM: 'medium'
};

var StyledButton = _styledComponents["default"].button(_templateObject3(), function (props) {
  return props.size === SIZES.SMALL ? '8px 16px' : '13px 20px';
}, function (props) {
  return props.size === SIZES.SMALL ? _styles.typography.size.s1 : _styles.typography.size.s2;
}, _styles.typography.weight.extrabold, function (props) {
  return !props.isLoading && "\n      &:hover {\n        transform: translate3d(0, -2px, 0);\n        box-shadow: rgba(0, 0, 0, 0.2) 0 2px 6px 0;\n      }\n\n      &:active {\n        transform: translate3d(0, 0, 0);\n      }\n\n      &:focus {\n        box-shadow: ".concat((0, _polished.rgba)(_styles.color.primary, 0.4), " 0 1px 9px 2px;\n      }\n\n      &:focus:hover {\n        box-shadow: ").concat((0, _polished.rgba)(_styles.color.primary, 0.2), " 0 8px 18px 0px;\n      }\n    ");
}, Text, _animation.easing.rubber, Loading, function (props) {
  return props.size === SIZES.SMALL ? '14' : '16';
}, function (props) {
  return props.size === SIZES.SMALL ? '14' : '16';
}, function (props) {
  return props.size === SIZES.SMALL ? '4' : '6';
}, function (props) {
  return props.size === SIZES.SMALL ? '-1' : '-2';
}, function (props) {
  return props.size === SIZES.SMALL ? '-1' : '-2';
}, function (props) {
  return props.disabled && "\n      cursor: not-allowed !important;\n      opacity: 0.5;\n      &:hover {\n        transform: none;\n      }\n    ";
}, function (props) {
  return props.isUnclickable && "\n      cursor: default !important;\n      pointer-events: none;\n      &:hover {\n        transform: none;\n      }\n    ";
}, function (props) {
  return props.isLoading && "\n      cursor: progress !important;\n      opacity: 0.7;\n\n      ".concat(Loading, " {\n        transition: transform 700ms ").concat(_animation.easing.rubber, ";\n        transform: translate3d(0, -50%, 0);\n        opacity: 1;\n      }\n\n      ").concat(Text, " {\n        transform: scale3d(0, 0, 1) translate3d(0, -100%, 0);\n        opacity: 0;\n      }\n\n      &:hover {\n        transform: none;\n      }\n    ");
}, function (props) {
  return props.containsIcon && "\n      svg {\n        display: block;\n        margin: 0;\n      }\n      padding: ".concat(props.size === SIZES.SMALL ? '7' : '12', "px;\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.PRIMARY && "\n      background: ".concat(_styles.color.primary, ";\n      color: ").concat(_styles.color.lightest, ";\n\n      ").concat(!props.isLoading && "\n          &:hover {\n            background: ".concat((0, _polished.darken)(0.05, _styles.color.primary), ";\n          }\n          &:active {\n            box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n          }\n          &:focus {\n            box-shadow: ").concat((0, _polished.rgba)(_styles.color.primary, 0.4), " 0 1px 9px 2px;\n          }\n          &:focus:hover {\n            box-shadow: ").concat((0, _polished.rgba)(_styles.color.primary, 0.2), " 0 8px 18px 0px;\n          }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.SECONDARY && "\n      background: ".concat(_styles.color.secondary, ";\n      color: ").concat(_styles.color.lightest, ";\n\n      ").concat(!props.isLoading && "\n          &:hover {\n            background: ".concat((0, _polished.darken)(0.05, _styles.color.secondary), ";\n          }\n          &:active {\n            box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n          }\n          &:focus {\n            box-shadow: ").concat((0, _polished.rgba)(_styles.color.secondary, 0.4), " 0 1px 9px 2px;\n          }\n          &:focus:hover {\n            box-shadow: ").concat((0, _polished.rgba)(_styles.color.secondary, 0.2), " 0 8px 18px 0px;\n          }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.TERTIARY && "\n      background: ".concat(_styles.color.tertiary, ";\n      color: ").concat(_styles.color.darkest, ";\n\n      ").concat(!props.isLoading && "\n          &:hover {\n            background: ".concat((0, _polished.darken)(0.05, _styles.color.tertiary), ";\n          }\n          &:active {\n            box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n          }\n          &:focus {\n            box-shadow: ").concat((0, _polished.rgba)(_styles.color.darkest, 0.15), " 0 1px 9px 2px;\n          }\n          &:focus:hover {\n            box-shadow: ").concat((0, _polished.rgba)(_styles.color.darkest, 0.05), " 0 8px 18px 0px;\n          }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.OUTLINE && "\n      box-shadow: ".concat((0, _polished.opacify)(0.05, _styles.color.border), " 0 0 0 1px inset;\n      color: ").concat(_styles.color.dark, ";\n      background: transparent;\n\n      ").concat(!props.isLoading && "\n          &:hover {\n            box-shadow: ".concat((0, _polished.opacify)(0.3, _styles.color.border), " 0 0 0 1px inset;\n          }\n\n          &:active {\n            background: ").concat((0, _polished.opacify)(0.05, _styles.color.border), ";\n            box-shadow: transparent 0 0 0 1px inset;\n            color: ").concat(_styles.color.darkest, ";\n          }\n\n          &:active:focus:hover {\n            ", "\n            background: ").concat((0, _polished.opacify)(0.05, _styles.color.border), ";\n            box-shadow:  ").concat((0, _polished.rgba)(_styles.color.darkest, 0.15), " 0 1px 9px 2px;\n          }\n\n          &:focus {\n            box-shadow: ").concat((0, _polished.opacify)(0.05, _styles.color.border), " 0 0 0 1px inset, \n            ").concat((0, _polished.rgba)(_styles.color.darkest, 0.15), " 0 1px 9px 2px;\n          }\n          &:focus:hover {\n            box-shadow: ").concat((0, _polished.opacify)(0.05, _styles.color.border), " 0 0 0 1px inset, \n            ").concat((0, _polished.rgba)(_styles.color.darkest, 0.05), " 0 8px 18px 0px;\n          }\n        "), ";\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.PRIMARY_OUTLINE && "\n      box-shadow: ".concat(_styles.color.primary, " 0 0 0 1px inset;\n      color: ").concat(_styles.color.primary, ";\n\n      &:hover {\n        box-shadow: ").concat(_styles.color.primary, " 0 0 0 1px inset;\n        background: transparent;\n      }\n\n      &:active {\n        background: ").concat(_styles.color.primary, ";\n        box-shadow: ").concat(_styles.color.primary, " 0 0 0 1px inset;\n        color: ").concat(_styles.color.lightest, ";\n      }\n      &:focus {\n        box-shadow: ").concat(_styles.color.primary, " 0 0 0 1px inset, ").concat((0, _polished.rgba)(_styles.color.primary, 0.4), " 0 1px 9px 2px;\n      }\n      &:focus:hover {\n        box-shadow: ").concat(_styles.color.primary, " 0 0 0 1px inset, ").concat((0, _polished.rgba)(_styles.color.primary, 0.2), " 0 8px 18px 0px;\n      }\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.SECONDARY_OUTLINE && "\n      box-shadow: ".concat(_styles.color.secondary, " 0 0 0 1px inset;\n      color: ").concat(_styles.color.secondary, ";\n\n      &:hover {\n        box-shadow: ").concat(_styles.color.secondary, " 0 0 0 1px inset;\n        background: transparent;\n      }\n\n      &:active {\n        background: ").concat(_styles.color.secondary, ";\n        box-shadow: ").concat(_styles.color.secondary, " 0 0 0 1px inset;\n        color: ").concat(_styles.color.lightest, ";\n      }\n      &:focus {\n        box-shadow: ").concat(_styles.color.secondary, " 0 0 0 1px inset,\n          ").concat((0, _polished.rgba)(_styles.color.secondary, 0.4), " 0 1px 9px 2px;\n      }\n      &:focus:hover {\n        box-shadow: ").concat(_styles.color.secondary, " 0 0 0 1px inset,\n          ").concat((0, _polished.rgba)(_styles.color.secondary, 0.2), " 0 8px 18px 0px;\n      }\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.INVERSE_PRIMARY && "\n        background: ".concat(_styles.color.lightest, ";\n        color: ").concat(_styles.color.primary, ";\n\n        ").concat(!props.isLoading && "\n            &:hover {\n              background: ".concat(_styles.color.lightest, ";\n            }\n            &:active {\n              box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n            }\n            &:focus {\n              box-shadow: ").concat((0, _polished.rgba)(_styles.color.primary, 0.4), " 0 1px 9px 2px;\n            }\n            &:focus:hover {\n              box-shadow: ").concat((0, _polished.rgba)(_styles.color.primary, 0.2), " 0 8px 18px 0px;\n            }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.INVERSE_SECONDARY && "\n        background: ".concat(_styles.color.lightest, ";\n        color: ").concat(_styles.color.secondary, ";\n\n        ").concat(!props.isLoading && "\n            &:hover {\n              background: ".concat(_styles.color.lightest, ";\n            }\n            &:active {\n              box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n            }\n            &:focus {\n              box-shadow: ").concat((0, _polished.rgba)(_styles.color.secondary, 0.4), " 0 1px 9px 2px;\n            }\n            &:focus:hover {\n              box-shadow: ").concat((0, _polished.rgba)(_styles.color.secondary, 0.2), " 0 8px 18px 0px;\n            }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.INVERSE_OUTLINE && "\n        box-shadow: ".concat(_styles.color.lightest, " 0 0 0 1px inset;\n        color: ").concat(_styles.color.lightest, ";\n\n        &:hover {\n          box-shadow: ").concat(_styles.color.lightest, " 0 0 0 1px inset;\n          background: transparent;\n        }\n\n        &:active {\n          background: ").concat(_styles.color.lightest, ";\n          box-shadow: ").concat(_styles.color.lightest, " 0 0 0 1px inset;\n          color: ").concat(_styles.color.darkest, ";\n        }\n        &:focus {\n          box-shadow: ").concat(_styles.color.lightest, " 0 0 0 1px inset,\n            ").concat((0, _polished.rgba)(_styles.color.darkest, 0.4), " 0 1px 9px 2px;\n        }\n        &:focus:hover {\n          box-shadow: ").concat(_styles.color.lightest, " 0 0 0 1px inset,\n            ").concat((0, _polished.rgba)(_styles.color.darkest, 0.2), " 0 8px 18px 0px;\n        }\n    ");
});

exports.StyledButton = StyledButton;

var UnstyledButton = _styledComponents["default"].button(_templateObject4());

var ButtonLink = _styledComponents["default"].a(_templateObject5()); // The main purpose of this component is to strip certain props that get passed
// down to the styled component, so that we don't end up passing them to a
// tag which would throw warnings for non-standard props.


var ButtonComponentPicker = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var appearance = _ref.appearance,
      ButtonWrapper = _ref.ButtonWrapper,
      containsIcon = _ref.containsIcon,
      isLink = _ref.isLink,
      isLoading = _ref.isLoading,
      isUnclickable = _ref.isUnclickable,
      size = _ref.size,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["appearance", "ButtonWrapper", "containsIcon", "isLink", "isLoading", "isUnclickable", "size"]);
  // Use the UnstyledButton here to avoid duplicating styles and creating
  // specificity conflicts by first rendering the StyledLink higher up the chain
  // and then re-rendering it through the 'as' prop.
  var ButtonComponent = isLink ? ButtonLink : ButtonWrapper || UnstyledButton;
  return /*#__PURE__*/_react["default"].createElement(ButtonComponent, rest);
});
var buttonStyleProps = {
  appearance: _propTypes["default"].oneOf(Object.values(APPEARANCES)),
  ButtonWrapper: _propTypes["default"].oneOfType([_propTypes["default"].func, _propTypes["default"].object]),

  /**
   Buttons with icons by themselves have a circular shape
  */
  containsIcon: _propTypes["default"].bool,

  /**
   Buttons that have hrefs should use <a> instead of <button>
  */
  isLink: _propTypes["default"].bool,
  isLoading: _propTypes["default"].bool,

  /**
   Prevents users from clicking on a button multiple times (for things like payment forms)
  */
  isUnclickable: _propTypes["default"].bool,
  size: _propTypes["default"].oneOf(Object.values(SIZES))
};
var buttonStyleDefaultProps = {
  isLoading: false,
  isLink: false,
  appearance: APPEARANCES.TERTIARY,
  isUnclickable: false,
  containsIcon: false,
  size: SIZES.MEDIUM,
  ButtonWrapper: undefined
};
ButtonComponentPicker.propTypes = _objectSpread({}, buttonStyleProps);
ButtonComponentPicker.defaultProps = _objectSpread({}, buttonStyleDefaultProps);
var Button = /*#__PURE__*/(0, _react.forwardRef)(function (_ref2, ref) {
  var children = _ref2.children,
      isDisabled = _ref2.isDisabled,
      isLoading = _ref2.isLoading,
      loadingText = _ref2.loadingText,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["children", "isDisabled", "isLoading", "loadingText"]);

  var content = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(Text, null, children), isLoading && /*#__PURE__*/_react["default"].createElement(Loading, null, loadingText || 'Loading...'));

  return /*#__PURE__*/_react["default"].createElement(StyledButton, (0, _extends2["default"])({
    as: ButtonComponentPicker,
    disabled: isDisabled,
    isLoading: isLoading,
    ref: ref
  }, rest), content);
});
exports.Button = Button;
Button.propTypes = _objectSpread(_objectSpread({}, buttonStyleProps), {}, {
  children: _propTypes["default"].node.isRequired,
  isDisabled: _propTypes["default"].bool,

  /**
   When a button is in the loading state you can supply custom text
  */
  loadingText: _propTypes["default"].node
});
Button.defaultProps = _objectSpread({
  loadingText: null,
  isDisabled: false
}, buttonStyleDefaultProps);
