// @storybook/design-system/dist/components/Checkbox.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Checkbox = Checkbox;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _polished = require("polished");

var _styles = require("./shared/styles");

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  align-items: center;\n  flex-wrap: wrap;\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  margin: 0 0.4em 0 0;\n  font-size: initial;\n  opacity: 0;\n  vertical-align: text-top;\n\n  & + ", " {\n    &:before,\n    &:after {\n      position: absolute;\n      top: 0;\n      left: 0;\n      height: 1em;\n      width: 1em;\n      content: '';\n      display: block;\n    }\n\n    &:before {\n      border-radius: 4px;\n    }\n\n    &:after {\n      border-radius: 3px;\n    }\n  }\n\n  & + ", ":before {\n    box-shadow: ", " 0 0 0 1px inset;\n  }\n\n  &:focus + ", ":before {\n    box-shadow: ", " 0 0 0 1px inset;\n  }\n\n  &:checked + ", ":before {\n    box-shadow: ", " 0 0 0 1px inset;\n  }\n\n  &:checked:focus + ", ":before {\n    box-shadow: ", " 0 0 0 1px inset,\n      ", " 0 0 5px 2px;\n  }\n\n  & + ", ":after {\n    transition: all 150ms ease-out;\n    transform: scale3d(0, 0, 1);\n\n    height: 10px;\n    margin-left: 2px;\n    margin-top: 2px;\n    width: 10px;\n\n    opacity: 0;\n  }\n\n  &:checked + ", ":after {\n    transform: scale3d(1, 1, 1);\n    background: ", ";\n    opacity: 1;\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-weight: ", ";\n  font-size: ", "px;\n  color: ", ";\n  margin-left: 6px;\n  height: 1em;\n  display: flex;\n  align-items: center;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      border: 0px !important;\n      clip: rect(0 0 0 0) !important;\n      -webkit-clip-path: inset(100%) !important;\n      clip-path: inset(100%) !important;\n      height: 1px !important;\n      overflow: hidden !important;\n      padding: 0px !important;\n      position: absolute !important;\n      white-space: nowrap !important;\n      width: 1px !important;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  cursor: pointer;\n  font-size: ", "px;\n  font-weight: ", ";\n  position: relative;\n  height: 1em;\n  display: flex;\n  align-items: center;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Label = _styledComponents["default"].label(_templateObject(), _styles.typography.size.s2, _styles.typography.weight.bold);

var OptionalText = _styledComponents["default"].span(_templateObject2(), function (props) {
  return props.hideLabel && (0, _styledComponents.css)(_templateObject3());
});

var Error = _styledComponents["default"].span(_templateObject4(), _styles.typography.weight.regular, _styles.typography.size.s2, _styles.color.negative);

var LabelText = _styledComponents["default"].span(_templateObject5());

var Input = _styledComponents["default"].input.attrs({
  type: 'checkbox'
})(_templateObject6(), LabelText, LabelText, _styles.color.mediumdark, LabelText, function (props) {
  return props.checkboxColor;
}, LabelText, function (props) {
  return props.checkboxColor;
}, LabelText, function (props) {
  return props.checkboxColor;
}, function (props) {
  return (0, _polished.rgba)(props.checkboxColor, 0.3);
}, LabelText, LabelText, function (props) {
  return props.checkboxColor;
});

var CheckboxWrapper = _styledComponents["default"].div(_templateObject7());

function Checkbox(_ref) {
  var appearance = _ref.appearance,
      id = _ref.id,
      label = _ref.label,
      error = _ref.error,
      hideLabel = _ref.hideLabel,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["appearance", "id", "label", "error", "hideLabel"]);
  var errorId = "".concat(id, "-error");
  var checkboxColor = _styles.color[appearance];
  return /*#__PURE__*/_react["default"].createElement(CheckboxWrapper, null, /*#__PURE__*/_react["default"].createElement(Label, null, /*#__PURE__*/_react["default"].createElement(Input, (0, _extends2["default"])({}, props, {
    id: id,
    "aria-describedby": errorId,
    "aria-invalid": !!error,
    checkboxColor: checkboxColor,
    type: "checkbox"
  })), /*#__PURE__*/_react["default"].createElement(LabelText, null, /*#__PURE__*/_react["default"].createElement(OptionalText, {
    hideLabel: hideLabel
  }, label))), /*#__PURE__*/_react["default"].createElement(Error, {
    id: errorId,
    error: error
  }, error));
}

Checkbox.propTypes = {
  appearance: _propTypes["default"].oneOf(['primary', 'secondary']),
  id: _propTypes["default"].string.isRequired,
  label: _propTypes["default"].string.isRequired,
  hideLabel: _propTypes["default"].bool,
  error: _propTypes["default"].string
};
Checkbox.defaultProps = {
  appearance: 'primary',
  hideLabel: false,
  error: null
};
