// @storybook/design-system/dist/components/CodeSnippets.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CodeSnippets = CodeSnippets;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Clipboard = require("./Clipboard");

var _Highlight = require("./Highlight");

var _styles = require("./shared/styles");

var _LinkTabs = require("./LinkTabs");

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  && {\n    transform: translateY(1px);\n    margin: 0;\n    padding: 0;\n\n    li {\n      margin: 0;\n    }\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  background: ", ";\n  border-top-left-radius: ", "px;\n  border-top-right-radius: ", "px;\n  border-bottom: 1px solid ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  bottom: -1px;\n  right: -1px;\n  border: 1px solid ", ";\n  border-radius: ", "px 0 0 0;\n  background-color: ", ";\n  font-size: ", "px;\n  font-weight: ", ";\n  letter-spacing: 0;\n  line-height: 14px;\n  padding: 5px 13px;\n\n  &:focus {\n    box-shadow: ", " 0 -3px 0 0 inset;\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: relative;\n  border-radius: ", "px;\n  overflow: hidden;\n\n  code {\n    font-size: ", "px;\n  }\n\n  && pre[class*='language-'] {\n    background-color: transparent;\n    padding: 1em;\n    padding-right: 2em;\n    margin: 0;\n    background: transparent;\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  background-color: #f7f9fc;\n  border: 1px solid ", ";\n  border-radius: ", "px;\n  box-shadow: 0 2px 5px 0 ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Wrapper = _styledComponents["default"].div(_templateObject());

var Background = _styledComponents["default"].div(_templateObject2(), _styles.color.border, _styles.spacing.borderRadius.small, _styles.color.border);

var StyledHighlight = (0, _styledComponents["default"])(_Highlight.Highlight)(_templateObject3(), _styles.spacing.borderRadius.small, _styles.typography.size.s2);
var StyledClipboard = (0, _styledComponents["default"])(_Clipboard.Clipboard)(_templateObject4(), _styles.color.border, _styles.spacing.borderRadius.small, _styles.color.lightest, _styles.typography.size.s1, _styles.typography.weight.bold, _styles.color.secondary);

function Snippet(_ref) {
  var snippet = _ref.snippet;
  var PreSnippetComponent = snippet.PreSnippet,
      SnippetComponent = snippet.Snippet;
  var snippetRef = (0, _react.useRef)();

  var getCopyContent = function getCopyContent() {
    return snippetRef.current && snippetRef.current.textContent;
  };

  return /*#__PURE__*/_react["default"].createElement(StyledHighlight, {
    withHTMLChildren: false
  }, PreSnippetComponent && /*#__PURE__*/_react["default"].createElement(PreSnippetComponent, null), /*#__PURE__*/_react["default"].createElement("div", {
    ref: snippetRef
  }, /*#__PURE__*/_react["default"].createElement(SnippetComponent, null)), /*#__PURE__*/_react["default"].createElement(StyledClipboard, {
    getCopyContent: getCopyContent
  }, function (copied) {
    return copied ? 'Copied' : 'Copy';
  }));
}

Snippet.propTypes = {
  snippet: _propTypes["default"].shape({
    Snippet: _propTypes["default"].elementType.isRequired,
    PreSnippet: _propTypes["default"].elementType
  }).isRequired
};

var TabsWrapper = _styledComponents["default"].div(_templateObject5(), _styles.color.lightest, _styles.spacing.borderRadius.small, _styles.spacing.borderRadius.small, _styles.color.border);

var StyledTabs = (0, _styledComponents["default"])(_LinkTabs.LinkTabs)(_templateObject6());

function SnippetList(_ref2) {
  var snippets = _ref2.snippets;

  var _useState = (0, _react.useState)(snippets[0]),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      activeSnippet = _useState2[0],
      setActiveSnippet = _useState2[1];

  var tabItems = snippets.map(function (snippet, index) {
    var isActive = activeSnippet.id === snippet.id;
    return {
      key: snippet.id,
      isActive: isActive,
      isButton: true,
      label: snippet.renderTabLabel({
        isActive: isActive
      }),
      onClick: function onClick() {
        return setActiveSnippet(snippets[index]);
      }
    };
  });
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(TabsWrapper, null, /*#__PURE__*/_react["default"].createElement(StyledTabs, {
    items: tabItems
  })), /*#__PURE__*/_react["default"].createElement(Snippet, {
    key: activeSnippet.id,
    snippet: activeSnippet
  }));
}

SnippetList.propTypes = {
  snippets: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    id: _propTypes["default"].string.isRequired,
    renderTabLabel: _propTypes["default"].func.isRequired
  }).isRequired).isRequired
};

function CodeSnippets(_ref3) {
  var snippets = _ref3.snippets,
      rest = (0, _objectWithoutProperties2["default"])(_ref3, ["snippets"]);
  return /*#__PURE__*/_react["default"].createElement(Wrapper, rest, /*#__PURE__*/_react["default"].createElement(Background, null, snippets.length === 1 ? /*#__PURE__*/_react["default"].createElement(Snippet, {
    snippet: snippets[0]
  }) : /*#__PURE__*/_react["default"].createElement(SnippetList, {
    snippets: snippets
  })));
}

CodeSnippets.propTypes = {
  snippets: _propTypes["default"].arrayOf(_propTypes["default"].shape({})).isRequired
};
