// @storybook/design-system/dist/components/Highlight.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Highlight = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _prismjs = _interopRequireDefault(require("prismjs"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("./shared/styles");

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  /*\n    Line below from: prismjs/themes/prism.css, with removal of overly specific\n    selectors (primarily [class*=language-]) and selectors that collide with our\n    global styling (primarly code & pre selectors).\n  */\n  .token.cdata,.token.comment,.token.doctype,.token.prolog{color:#708090}.token.punctuation{color:#999}.namespace{opacity:.7}.token.boolean,.token.constant,.token.deleted,.token.number,.token.property,.token.symbol,.token.tag{color:#905}.token.attr-name,.token.builtin,.token.char,.token.inserted,.token.selector,.token.string{color:#690}.language-css .token.string,.style .token.string,.token.entity,.token.operator,.token.url{color:#a67f59;background:hsla(0,0%,100%,.5)}.token.atrule,.token.attr-value,.token.keyword{color:#07a}.token.class-name,.token.function{color:#dd4a68}.token.important,.token.regex,.token.variable{color:#e90}.token.bold,.token.important{font-weight:700}.token.italic{font-style:italic}.token.entity{cursor:help}\n\n  code, pre {\n    color: ", ";\n  }\n\n  code {\n    white-space: pre;\n  }\n\n  pre {\n    padding: 11px 1rem;\n    margin: 1rem 0;\n    background: ", ";\n    overflow: auto;\n  }\n\n  .language-bash .token.operator,\n  .language-bash .token.function,\n  .language-bash .token.builtin {\n    color: ", ";\n    background: none;\n  }\n\n  .language-markdown .token.title,\n  .language-md .token.title {\n    color: #dd4a68;\n  }\n\n  .language-markdown .token.table-header,\n  .language-md .token.table-header {\n    color: #07a;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

if (typeof document !== 'undefined') {
  global.Prism = _prismjs["default"];

  require('prismjs/components/prism-bash');

  require('prismjs/components/prism-javascript');

  require('prismjs/components/prism-typescript');

  require('prismjs/components/prism-json');

  require('prismjs/components/prism-css');

  require('prismjs/components/prism-yaml');

  require('prismjs/components/prism-markdown');

  require('prismjs/components/prism-jsx');

  require('prismjs/components/prism-tsx');
}

var languages = ['bash', 'javascript', 'typescript', 'json', 'css', 'yaml', 'markdown', 'md', 'jsx', 'tsx']; // Prism theme copied from 'prismjs/themes/prism.css.' -- without Webpack, the CSS
// cannot be imported easily and any app which pulls in the design system will
// need to handle the CSS loading itself. Therefore, it is easiest to just copy
// the theme over.
// prettier-ignore

var HighlightBlock = _styledComponents["default"].div(_templateObject(), _styles.color.darkest, _styles.color.lighter, _styles.color.darkest);

var languageMap = {
  mdx: 'markdown'
};

var Highlight = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Highlight, _React$Component);

  var _super = _createSuper(Highlight);

  function Highlight() {
    (0, _classCallCheck2["default"])(this, Highlight);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(Highlight, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.highlightCode();
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate() {
      this.highlightCode();
    }
  }, {
    key: "highlightCode",
    value: function highlightCode() {
      var domNode = _reactDom["default"].findDOMNode(this); // eslint-disable-line


      _prismjs["default"].highlightAllUnder(domNode);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          inputLanguage = _this$props.language,
          withHTMLChildren = _this$props.withHTMLChildren,
          rest = (0, _objectWithoutProperties2["default"])(_this$props, ["children", "language", "withHTMLChildren"]);
      var language = languageMap[inputLanguage] || inputLanguage;
      var codeBlock = withHTMLChildren ? /*#__PURE__*/_react["default"].createElement("div", {
        dangerouslySetInnerHTML: {
          __html: children
        }
      }) : children;
      return /*#__PURE__*/_react["default"].createElement(HighlightBlock, rest, language ? /*#__PURE__*/_react["default"].createElement("pre", {
        className: "language-".concat(language)
      }, /*#__PURE__*/_react["default"].createElement("code", {
        className: "language-".concat(language)
      }, codeBlock)) : codeBlock);
    }
  }]);
  return Highlight;
}(_react["default"].Component);

exports.Highlight = Highlight;
Highlight.propTypes = {
  children: _propTypes["default"].node.isRequired,
  language: _propTypes["default"].oneOf(languages),
  withHTMLChildren: _propTypes["default"].bool
};
Highlight.defaultProps = {
  language: null,
  withHTMLChildren: true
};
