// @storybook/design-system/dist/components/Input.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Input = Input;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styles = require("./shared/styles");

var _animation = require("./shared/animation");

var _Icon = require("./Icon");

function _templateObject21() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    display: table-row;\n\n    ", ", ", " {\n      display: table-cell;\n    }\n\n    ", " {\n      width: 1px;\n      padding-right: 20px;\n      vertical-align: middle;\n    }\n\n    ", " {\n      width: auto;\n    }\n\n  "]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n"]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", " {\n          box-shadow: ", " 0 0 0 1px inset;\n          &:focus { box-shadow: ", " 0 0 0 1px inset !important;  }\n      }\n    "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", " {\n        opacity: 0;\n      }\n      ", ":hover + ", ",\n      ", ":focus + ", " {\n        transform: translate3d(0%, -100%, 0);\n        opacity: 1;\n        padding: .25em 1.25em .25em .5em;\n      }\n    "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", " {\n        opacity: 0;\n        transform: translate3d(100%, -50%, 0);\n      }\n    "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    ", " {\n      color: ", ";\n      background: none;\n      transform: translate3d(0%, -50%, 0);\n      opacity: 1;\n      padding: .25em 1.25em .25em .5em;\n    }\n\n    ", ":hover + ", ",\n    ", ":focus + ", " {\n      opacity: 0;\n      transform: translate3d(100%, -50%, 0);\n      padding: 0;\n    }\n\n    ", "\n\n    ", "\n\n    ", ";\n\n    svg {\n      animation: ", " 700ms ease-out;\n      path { fill: ", "; }\n    }\n  "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = (0, _taggedTemplateLiteral2["default"])([" padding-left: 1.75em; "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = (0, _taggedTemplateLiteral2["default"])([" padding-left: 2.4em; "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    > svg {\n      transition: all 150ms ease-out;\n      position: absolute;\n      top: 50%;\n      height: 1em;\n      width: 1em;\n  \t\tfont-size: ", "em;\n  \t\tmargin-top: -.5em;\n  \t\tz-index: 1;\n\n      background: transparent;\n\n      path {\n        transition: all 150ms ease-out;\n        fill: ", ";\n      }\n    }\n\n    ", ":focus + svg path {\n      fill: ", ";\n    }\n\n    ", " {\n      padding-left: 2.75em;\n\n      ", ";\n      ", ";\n    }\n    > svg { left: ", "em; }\n\n  "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      top: -4px;\n      right: auto;\n      left: 0;\n      border-radius: ", "px;\n      padding: 6px;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = (0, _taggedTemplateLiteral2["default"])([" right: 0; "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      font-size: ", "px;\n      line-height: 16px;\n      font-family: ", ";\n      border-radius: 2px;\n      background: rgba(0,0,0,.05);\n      padding: 3px 6px;\n\n      &:focus { box-shadow: ", " 0 0 0 1px inset; }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      font-size: ", "px;\n      line-height: 16px;\n      padding: .5em 1em; //28\n      border-radius: 3em;\n      background: transparent;\n\n      box-shadow: ", " 0 0 0 1px inset;\n\n      &:focus { box-shadow: ", " 0 0 0 1px inset; }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      padding: 0;\n      border: none;\n      box-shadow: none;\n      background: none;\n\n      &:focus { box-shadow: none !important; }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      box-shadow: ", " 0 0 0 1px inset;\n\n      &:focus { box-shadow: ", " 0 0 0 1px inset; }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  position: relative;\n  vertical-align: top;\n  width: 100%;\n\n  ", " {\n    background: ", ";\n    border-radius: 0;\n    color: ", ";\n    font-family: ", ";\n    font-size: ", "px;\n    line-height: 20px;\n    padding: .715em 1em; //40\n\n    &:focus { box-shadow: ", " 0 0 0 1px inset; }\n\n    ", "\n\n    ", "\n\n    ", "\n\n    ", "\n  }\n\n  ", " {\n    position: absolute;\n    top: 50%;\n    right: 1px;\n    margin-left: 1px;\n    transform: translate3d(100%, -50%, 0);\n    transition: all 200ms ease-out;\n    font-family: ", " ;\n    font-size: ", "px;\n    line-height: 1em;\n    opacity: 0;\n    pointer-events: none;\n\n    background: ", ";\n    color: ", ";\n\n    ", "\n    ", "\n  }\n\n  ", "\n\n  ", "\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  right: 0;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ::placeholder {\n    color: ", ";\n    font-weight: ", ";\n  }\n  appearance: none;\n  border:none;\n  box-sizing: border-box;\n  display: block;\n  outline: none;\n  width: 100%;\n  margin: 0;\n\n  &[disabled] {\n    cursor: not-allowed;\n    opacity: .5;\n  }\n\n  &:-webkit-autofill { -webkit-box-shadow: 0 0 0 3em ", " inset; }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    border: 0px !important;\n    clip: rect(0 0 0 0) !important;\n    -webkit-clip-path: inset(100%) !important;\n    clip-path: inset(100%) !important;\n    height: 1px !important;\n    overflow: hidden !important;\n    padding: 0px !important;\n    position: absolute !important;\n    white-space: nowrap !important;\n    width: 1px !important;\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  margin-bottom: 0.33em;\n  \n  ", "\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-weight: ", ";\n  font-family: ", ";\n  font-size: ", "px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

// prettier-ignore
var Label = _styledComponents["default"].label(_templateObject(), function (props) {
  return props.appearance !== 'code' && _styles.typography.weight.bold;
}, function (props) {
  return props.appearance === 'code' && _styles.typography.type.code;
}, function (props) {
  return props.appearance === 'code' ? _styles.typography.size.s1 : _styles.typography.size.s2;
}); // prettier-ignore


var LabelWrapper = _styledComponents["default"].div(_templateObject2(), function (props) {
  return props.hideLabel && (0, _styledComponents.css)(_templateObject3());
}); // prettier-ignore


var InputText = _styledComponents["default"].input.attrs({
  type: 'text'
})(_templateObject4(), _styles.color.mediumdark, _styles.typography.weight.bold, _styles.color.lightest);

var Error = _styledComponents["default"].div(_templateObject5()); // prettier-ignore


var InputWrapper = _styledComponents["default"].div(_templateObject6(), InputText, _styles.color.lightest, _styles.color.darkest, function (props) {
  return props.appearance === 'code' && _styles.typography.type.code;
}, function (props) {
  return props.appearance === 'code' ? _styles.typography.size.s1 : _styles.typography.size.s2;
}, _styles.color.primary, function (props) {
  return props.appearance === 'secondary' && (0, _styledComponents.css)(_templateObject7(), _styles.color.mediumlight, _styles.color.secondary);
}, function (props) {
  return props.appearance === 'tertiary' && (0, _styledComponents.css)(_templateObject8());
}, function (props) {
  return props.appearance === 'pill' && (0, _styledComponents.css)(_templateObject9(), _styles.typography.size.s1, _styles.color.medium, _styles.color.secondary);
}, function (props) {
  return props.appearance === 'code' && (0, _styledComponents.css)(_templateObject10(), _styles.typography.size.s2 - 1, _styles.typography.type.code, _styles.color.secondary);
}, Error, function (props) {
  return props.appearance === 'code' ? _styles.typography.type.code : _styles.typography.type.primary;
}, _styles.typography.size.s1, function (props) {
  return props.appearance !== 'tertiary' && 'rgba(255,255,255,.9)';
}, _styles.color.negative, function (props) {
  return props.appearance === 'tertiary' && (0, _styledComponents.css)(_templateObject11());
}, function (props) {
  return props.appearance === 'code' && (0, _styledComponents.css)(_templateObject12(), _styles.spacing.borderRadius.small);
}, function (props) {
  return props.icon && (0, _styledComponents.css)(_templateObject13(), props.appearance === 'pill' ? 0.75 : 1, _styles.color.mediumdark, InputText, _styles.color.darker, InputText, props.appearance === 'pill' && (0, _styledComponents.css)(_templateObject14()), props.appearance === 'tertiary' && (0, _styledComponents.css)(_templateObject15()), props.appearance === 'tertiary' ? 0 : 0.8);
}, function (props) {
  return props.error && (0, _styledComponents.css)(_templateObject16(), Error, _styles.color.negative, InputText, Error, InputText, Error, props.focused && (0, _styledComponents.css)(_templateObject17(), Error), props.appearance === 'code' && (0, _styledComponents.css)(_templateObject18(), Error, InputText, Error, InputText, Error), props.appearance !== 'tertiary' && (0, _styledComponents.css)(_templateObject19(), InputText, _styles.color.negative, _styles.color.negative), _animation.jiggle, _styles.color.negative);
}); // prettier-ignore


var InputContainer = _styledComponents["default"].div(_templateObject20(), function (props) {
  return props.orientation === 'horizontal' && (0, _styledComponents.css)(_templateObject21(), LabelWrapper, InputWrapper, LabelWrapper, InputWrapper);
});

function Input(_ref) {
  var id = _ref.id,
      value = _ref.value,
      label = _ref.label,
      hideLabel = _ref.hideLabel,
      orientation = _ref.orientation,
      icon = _ref.icon,
      error = _ref.error,
      appearance = _ref.appearance,
      className = _ref.className,
      focused = _ref.focused,
      lastErrorValue = _ref.lastErrorValue,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["id", "value", "label", "hideLabel", "orientation", "icon", "error", "appearance", "className", "focused", "lastErrorValue"]);
  var errorId = "".concat(id, "-error");
  var errorMessage = error;

  if (lastErrorValue) {
    if (value !== lastErrorValue) {
      errorMessage = null;
    }
  }

  return /*#__PURE__*/_react["default"].createElement(InputContainer, {
    orientation: orientation,
    className: className
  }, /*#__PURE__*/_react["default"].createElement(LabelWrapper, {
    hideLabel: hideLabel
  }, /*#__PURE__*/_react["default"].createElement(Label, {
    htmlFor: id,
    appearance: appearance
  }, label)), /*#__PURE__*/_react["default"].createElement(InputWrapper, {
    error: errorMessage,
    "data-error": error,
    icon: icon,
    appearance: appearance,
    focused: focused
  }, icon && /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: icon,
    "aria-hidden": true
  }), /*#__PURE__*/_react["default"].createElement(InputText, (0, _extends2["default"])({
    id: id,
    value: value,
    "aria-describedby": errorId,
    "aria-invalid": !!error
  }, props)), /*#__PURE__*/_react["default"].createElement(Error, {
    id: errorId
  }, error)));
}

Input.propTypes = {
  id: _propTypes["default"].string.isRequired,
  value: _propTypes["default"].string,
  appearance: _propTypes["default"].oneOf(['default', 'secondary', 'tertiary', 'pill', 'code']),
  label: _propTypes["default"].string.isRequired,
  hideLabel: _propTypes["default"].bool,
  orientation: _propTypes["default"].oneOf(['vertical', 'horizontal']),
  icon: _propTypes["default"].string,
  error: _propTypes["default"].string,
  className: _propTypes["default"].string,
  focused: _propTypes["default"].bool,
  lastErrorValue: _propTypes["default"].string
};
Input.defaultProps = {
  value: '',
  appearance: 'default',
  hideLabel: false,
  orientation: 'vertical',
  icon: null,
  error: null,
  className: null,
  focused: false,
  lastErrorValue: null
};
