// @storybook/design-system/dist/components/ProgressDots.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ProgressDots = ProgressDots;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styles = require("./shared/styles");

var _animation = require("./shared/animation");

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      animation: ", " 1.5s ease-in-out infinite;\n      &:nth-child(1) {\n        animation-delay: 0s;\n      }\n      &:nth-child(2) {\n        animation-delay: 0.3s;\n      }\n      &:nth-child(3) {\n        animation-delay: 0.6s;\n      }\n      &:nth-child(4) {\n        animation-delay: 0.9s;\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      background: ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  background: ", ";\n  display: inline-block;\n  vertical-align: top;\n  width: ", "px;\n  height: ", "px;\n  border-radius: 3em;\n  margin: 0 ", "px;\n\n  ", ";\n\n  ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  height: ", "px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var ProgressWrapper = _styledComponents["default"].div(_templateObject(), function (props) {
  return props.size === 'small' ? 4 : 5;
});

var Dot = _styledComponents["default"].div(_templateObject2(), _styles.color.medium, function (props) {
  return props.size === 'small' ? 4 : 5;
}, function (props) {
  return props.size === 'small' ? 4 : 5;
}, function (props) {
  return props.size === 'small' ? 2 : 3;
}, function (props) {
  return props.active && (0, _styledComponents.css)(_templateObject3(), _styles.color.mediumdark);
}, function (props) {
  return props.isLoading && (0, _styledComponents.css)(_templateObject4(), _animation.glow);
});

function ProgressDots(_ref) {
  var isLoading = _ref.isLoading,
      steps = _ref.steps,
      progress = _ref.progress,
      className = _ref.className,
      size = _ref.size;
  var dots = [];

  for (var i = 0; i < steps; i += 1) {
    dots.push( /*#__PURE__*/_react["default"].createElement(Dot, {
      isLoading: isLoading,
      active: i === progress - 1,
      key: i,
      size: size
    }));
  }

  return /*#__PURE__*/_react["default"].createElement(ProgressWrapper, {
    className: className,
    size: size,
    role: "status",
    "aria-live": "polite",
    "aria-label": "Content is loading ..."
  }, dots);
}

ProgressDots.propTypes = {
  isLoading: _propTypes["default"].bool,
  steps: _propTypes["default"].number,
  progress: _propTypes["default"].number,
  className: _propTypes["default"].string,
  size: _propTypes["default"].oneOf(['small', 'large'])
};
ProgressDots.defaultProps = {
  isLoading: false,
  steps: 4,
  progress: 0,
  className: null,
  size: 'small'
};
