// @storybook/design-system/dist/components/Select.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Select = Select;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styles = require("./shared/styles");

var _animation = require("./shared/animation");

var _Icon = require("./Icon");

var _Spinner = require("./Spinner");

function _templateObject17() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    ", " {\n      box-shadow: ", " 0 0 0 1px inset;\n      &:focus { box-shadow: ", " 0 0 0 1px inset !important;  }\n    }\n\n    ", " + ", " {\n      animation: ", " 700ms ease-out;\n      path { fill: ", "; }\n    }\n  "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    ", " { padding-left: 2.5em; }\n\n    ", " + ", " {\n      transition: all 150ms ease-out ;\n      position: absolute;\n      top: 50%;\n      left: .8em;\n\t\t\theight: 1em;\n      width: 1em;\n\t\t\tmargin-top: -.5em;\n\t\t\tz-index: 1;\n\n\t    path { fill: ", "; }\n    }\n    ", ":focus + ", " path {\n      fill: ", ";\n\t  }\n  "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      width: auto;\n      height: auto;\n\n      ", " {\n        color: ", ";\n        background-color: transparent;\n        font-weight: ", ";\n        padding: 0 1.5em 0 0;\n        text-decoration: none;\n\n        &:focus {\n          box-shadow: none !important;\n        }\n      }\n      &:before {\n        content: none;\n      }\n      ", " {\n        right: 0;\n      }\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", " {\n        box-shadow: ", " 0 0 0 1px inset;\n      }\n      ", ":focus {\n        box-shadow: ", " 0 0 0 1px inset;\n      }\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      opacity: 0.5;\n    "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  height: 40px;\n  line-height: normal;\n  overflow: hidden;\n  position: relative;\n  vertical-align: top;\n  width: 100%;\n\n  transition: all 150ms ease-out;\n  transform: translate3d(0,0,0);\n\n  &:hover {\n    transform: translate3d(0,-1px,0);\n  }\n\n  &:active {\n    transform: translate3d(0,0,0);\n  }\n\n  &:before {\n    content: \"\";\n\t  bottom: 1px;\n\t\tright: 1px;\n\t\ttop: 1px;\n\t\twidth: 2em;\n    margin-left: 1px;\n\t\tposition: absolute;\n\t\tz-index: 1;\n\t\tpointer-events: none;\n    border-radius: ", "px;\n\t}\n\n  ", " {\n    position: absolute;\n\t\tz-index: 1;\n\t\tpointer-events: none;\n    height: 12px;\n    margin-top: -6px;\n\t\tright: 12px;\n\t\ttop: 50%;\n\n    path {fill: ", " }\n  }\n\n  ", ":focus { box-shadow: ", " 0 0 0 1px inset; }\n\n  ", "\n\n  &:before { background-color: rgba(255,255,255,.9); }\n\t", " {\n    background-color: ", ";\n    color: ", ";\n  }\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  right: 0;\n  top: 0;\n\n  transition: all 300ms cubic-bezier(0.175, 0.885, 0.335, 1.05);\n  font-size: ", "px;\n  font-family: ", ";\n\n  color: ", ";\n  line-height: 38px;\n  padding-right: 2.75em;\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  right: 16px;\n  left: auto;\n  z-index: 1;\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      cursor: progress;\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      cursor: not-allowed;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  appearance: none;\n  border: 0;\n  border-radius: 0;\n  font-size: ", "px;\n  line-height: 20px;\n  padding: 10px 3em 10px 1em;\n  position: relative;\n  outline: none;\n  width: 100%;\n\n  ", "\n\n  ", "\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      border: 0px !important;\n      clip: rect(0 0 0 0) !important;\n      -webkit-clip-path: inset(100%) !important;\n      clip-path: inset(100%) !important;\n      height: 1px !important;\n      overflow: hidden !important;\n      padding: 0px !important;\n      position: absolute !important;\n      white-space: nowrap !important;\n      width: 1px !important;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  margin-bottom: 0.5em;\n\n  ", "\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-weight: ", ";\n  font-size: ", "px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Label = _styledComponents["default"].label(_templateObject(), _styles.typography.weight.bold, _styles.typography.size.s2);

var LabelWrapper = _styledComponents["default"].div(_templateObject2(), function (props) {
  return props.hideLabel && (0, _styledComponents.css)(_templateObject3());
});

var Selector = _styledComponents["default"].select(_templateObject4(), _styles.typography.size.s2, function (props) {
  return props.disabled && (0, _styledComponents.css)(_templateObject5());
}, function (props) {
  return props.inProgress && (0, _styledComponents.css)(_templateObject6());
});

var OptionWrapper = _styledComponents["default"].option(_templateObject7());

var Arrow = (0, _styledComponents["default"])(_Icon.Icon).attrs({
  icon: 'arrowdown'
})(_templateObject8());
var SelectIcon = (0, _styledComponents["default"])(_Icon.Icon)(_templateObject9());
var SelectSpinner = (0, _styledComponents["default"])(_Spinner.Spinner)(_templateObject10());

var SelectError = _styledComponents["default"].div(_templateObject11(), _styles.typography.size.s1, _styles.typography.type.primary, _styles.color.negative);

var SelectWrapper = _styledComponents["default"].span(_templateObject12(), _styles.spacing.borderRadius.small, Arrow, _styles.color.mediumdark, Selector, _styles.color.primary, function (props) {
  return props.disabled && (0, _styledComponents.css)(_templateObject13());
}, Selector, _styles.color.lightest, _styles.color.darkest, function (props) {
  return props.appearance === 'secondary' && (0, _styledComponents.css)(_templateObject14(), Selector, _styles.color.mediumlight, Selector, _styles.color.secondary);
}, function (props) {
  return props.appearance === 'tertiary' && (0, _styledComponents.css)(_templateObject15(), Selector, _styles.color.dark, _styles.typography.weight.bold, Arrow);
}, function (props) {
  return props.icon && (0, _styledComponents.css)(_templateObject16(), Selector, Selector, SelectIcon, _styles.color.mediumdark, Selector, SelectIcon, _styles.color.darker);
}, function (props) {
  return props.error && (0, _styledComponents.css)(_templateObject17(), Selector, _styles.color.negative, _styles.color.negative, Selector, SelectIcon, _animation.jiggle, _styles.color.negative);
});

function Option(_ref) {
  var label = _ref.label,
      value = _ref.value;
  return /*#__PURE__*/_react["default"].createElement(OptionWrapper, {
    value: value
  }, label);
}

Option.propTypes = {
  label: _propTypes["default"].string.isRequired,
  value: _propTypes["default"].string.isRequired
};

function Select(_ref2) {
  var id = _ref2.id,
      options = _ref2.options,
      value = _ref2.value,
      appearance = _ref2.appearance,
      label = _ref2.label,
      hideLabel = _ref2.hideLabel,
      error = _ref2.error,
      icon = _ref2.icon,
      className = _ref2.className,
      inProgress = _ref2.inProgress,
      disabled = _ref2.disabled,
      other = (0, _objectWithoutProperties2["default"])(_ref2, ["id", "options", "value", "appearance", "label", "hideLabel", "error", "icon", "className", "inProgress", "disabled"]);
  var spinnerId;
  var errorId;
  var ariaDescribedBy;

  if (inProgress) {
    spinnerId = "".concat(id, "-in-progress");
    ariaDescribedBy = spinnerId;
  }

  if (error) {
    errorId = "".concat(id, "-error");
    ariaDescribedBy = "".concat(ariaDescribedBy || '', " ").concat(errorId);
  }

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: className
  }, /*#__PURE__*/_react["default"].createElement(LabelWrapper, {
    hideLabel: hideLabel
  }, /*#__PURE__*/_react["default"].createElement(Label, {
    htmlFor: id
  }, label)), /*#__PURE__*/_react["default"].createElement(SelectWrapper, {
    appearance: appearance,
    icon: icon,
    error: error,
    "data-error": error,
    disabled: disabled
  }, !inProgress && /*#__PURE__*/_react["default"].createElement(Arrow, null), /*#__PURE__*/_react["default"].createElement(Selector, (0, _extends2["default"])({
    id: id,
    value: value
  }, other, {
    disabled: disabled || inProgress,
    inProgress: inProgress,
    "aria-describedby": ariaDescribedBy,
    "aria-invalid": !!error,
    "aria-busy": inProgress
  }), options.map(function (option) {
    return /*#__PURE__*/_react["default"].createElement(Option, (0, _extends2["default"])({}, option, {
      key: option.value
    }));
  })), icon && /*#__PURE__*/_react["default"].createElement(SelectIcon, {
    icon: icon,
    "aria-hidden": true
  }), error && /*#__PURE__*/_react["default"].createElement(SelectError, {
    id: errorId
  }, error), inProgress && /*#__PURE__*/_react["default"].createElement(SelectSpinner, {
    id: spinnerId,
    "aria-label": "Loading",
    inForm: true
  })));
}

Select.propTypes = {
  id: _propTypes["default"].string.isRequired,
  options: _propTypes["default"].arrayOf(_propTypes["default"].shape(Option.propTypes)),
  value: _propTypes["default"].string,
  appearance: _propTypes["default"].oneOf(['default', 'secondary', 'tertiary']),
  label: _propTypes["default"].string.isRequired,
  hideLabel: _propTypes["default"].bool,
  error: _propTypes["default"].string,
  icon: _propTypes["default"].string,
  className: _propTypes["default"].string,
  inProgress: _propTypes["default"].bool,
  disabled: _propTypes["default"].bool
};
Select.defaultProps = {
  value: 'loading',
  options: [{
    title: 'Loading',
    value: 'loading'
  }],
  appearance: 'default',
  hideLabel: false,
  error: null,
  icon: null,
  className: null,
  inProgress: false,
  disabled: false
};
