// @storybook/design-system/dist/components/StoryLinkWrapper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StoryLinkWrapper = StoryLinkWrapper;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _addonActions = require("@storybook/addon-actions");

/* eslint-disable import/no-extraneous-dependencies */
// This is allows us to test whether the link works via the actions addon
var fireClickAction = (0, _addonActions.action)('onLinkClick');

function StoryLinkWrapper(_ref) {
  var children = _ref.children,
      className = _ref.className,
      href = _ref.href,
      onClick = _ref.onClick,
      to = _ref.to,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["children", "className", "href", "onClick", "to"]);

  var modifiedOnClick = function modifiedOnClick(event) {
    event.preventDefault();
    onClick();
    fireClickAction(href || to);
  };

  return /*#__PURE__*/_react["default"].createElement("a", (0, _extends2["default"])({
    className: className,
    href: href || to,
    onClick: modifiedOnClick
  }, rest), children);
}

StoryLinkWrapper.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  children: _propTypes["default"].any.isRequired,
  className: _propTypes["default"].string,
  href: _propTypes["default"].string,
  onClick: _propTypes["default"].func,
  to: _propTypes["default"].string
};
StoryLinkWrapper.defaultProps = {
  className: '',
  href: null,
  onClick: function onClick() {},
  to: null
};
