// @storybook/design-system/dist/components/modal/Modal.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Modal = Modal;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = require("prop-types");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactModal = _interopRequireDefault(require("react-modal"));

var _Button = require("../Button");

var _Icon = require("../Icon");

var _styles = require("../shared/styles");

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  position: absolute;\n  top: 0;\n  bottom: 0;\n  right: 0;\n  left: 0;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", ";\n  width: 100%;\n  /* Ensures overflow is not cutoff bc flexbox centering */\n  margin-top: auto !important;\n  margin-bottom: auto !important;\n  padding-top: 3rem;\n  padding-bottom: 3rem;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  z-index: 2;\n\n  bottom: 10vh;\n  left: 50%;\n  transform: translateX(-50%);\n\n  @media (min-width: ", "px) {\n    right: 30px;\n    top: 30px;\n    bottom: auto;\n    left: auto;\n    transform: none;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Action = _styledComponents["default"].div(_templateObject(), _styles.breakpoint);

var CenteredItem = _styledComponents["default"].div(_templateObject2(), _styles.pageMargins);

var CenteredWrapper = _styledComponents["default"].div(_templateObject3());

function Modal(_ref) {
  var isOpen = _ref.isOpen,
      onClose = _ref.onClose,
      children = _ref.children;
  var actions = {
    onClose: onClose
  };
  return /*#__PURE__*/_react["default"].createElement(_reactModal["default"], {
    isOpen: isOpen,
    onRequestClose: onClose,
    contentLabel: "Example Modal",
    ariaHideApp: false,
    closeTimeoutMS: 300,
    style: {
      overlay: {
        backgroundColor: 'rgba(246, 249, 252, .97)',
        // 1 less that max, to allow intercom over the top
        zIndex: 2147483646,
        overflowX: 'hidden',
        overflowY: 'auto'
      },
      content: {
        left: '0',
        top: '0',
        bottom: '0',
        right: '0',
        backgroundColor: 'transparent',
        border: 'none',
        overflow: 'visible',
        padding: '0',
        width: '100%',
        margin: 'auto'
      }
    }
  }, /*#__PURE__*/_react["default"].createElement(CenteredWrapper, null, /*#__PURE__*/_react["default"].createElement(CenteredItem, null, _react.Children.only(children(actions)))), /*#__PURE__*/_react["default"].createElement(Action, null, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    containsIcon: true,
    appearance: "outline",
    onClick: onClose
  }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "cross"
  }))));
}

Modal.propTypes = {
  isOpen: _propTypes.PropTypes.bool.isRequired,
  onClose: _propTypes.PropTypes.func.isRequired,
  children: _propTypes.PropTypes.func.isRequired
};
