// @workday/canvas-kit-react-status-indicator/dist/es6/lib/StatusIndicator.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { SystemIcon } from '@workday/canvas-kit-react-icon';
import { borderRadius, colors, type, spacing } from '@workday/canvas-kit-react-core';
import styled from '@emotion/styled';
export var StatusIndicatorType;
(function (StatusIndicatorType) {
    StatusIndicatorType["Gray"] = "gray";
    StatusIndicatorType["Orange"] = "orange";
    StatusIndicatorType["Blue"] = "blue";
    StatusIndicatorType["Green"] = "green";
    StatusIndicatorType["Red"] = "red";
    StatusIndicatorType["Transparent"] = "transparent";
})(StatusIndicatorType || (StatusIndicatorType = {}));
export var StatusIndicatorEmphasis;
(function (StatusIndicatorEmphasis) {
    StatusIndicatorEmphasis["High"] = "high";
    StatusIndicatorEmphasis["Low"] = "low";
})(StatusIndicatorEmphasis || (StatusIndicatorEmphasis = {}));
export var statusIndicatorStyles = {
    classname: 'status-indicator',
    styles: __assign(__assign({}, type.small), { display: 'inline-flex', alignItems: 'center', verticalAlign: 'middle', maxWidth: 150, height: spacing.s, padding: "1px " + spacing.xxxs, borderRadius: borderRadius.s, boxSizing: 'border-box' }),
    variants: {
        gray: {
            high: {
                color: colors.frenchVanilla100,
                background: colors.licorice300,
            },
            low: {
                color: colors.licorice400,
                background: colors.soap200,
            },
        },
        orange: {
            high: {
                color: colors.licorice500,
                background: colors.cantaloupe400,
            },
            low: {
                color: colors.toastedMarshmallow600,
                background: colors.cantaloupe100,
            },
        },
        blue: {
            high: {
                color: colors.frenchVanilla100,
                background: colors.blueberry400,
            },
            low: {
                color: colors.blueberry500,
                background: colors.blueberry100,
            },
        },
        green: {
            high: {
                color: colors.frenchVanilla100,
                background: colors.greenApple600,
            },
            low: {
                color: colors.greenApple600,
                background: colors.greenApple100,
            },
        },
        red: {
            high: {
                color: colors.frenchVanilla100,
                background: colors.cinnamon500,
            },
            low: {
                color: colors.cinnamon600,
                background: colors.cinnamon100,
            },
        },
        transparent: {
            high: {
                color: colors.frenchVanilla100,
                background: colors.blackPepper600,
            },
            low: {
                color: colors.frenchVanilla100,
                background: colors.blackPepper600,
            },
        },
    },
};
var Container = styled('span')(statusIndicatorStyles.styles, function (_a) {
    var type = _a.type, emphasis = _a.emphasis;
    return (__assign({}, statusIndicatorStyles.variants[type][emphasis]));
});
var StatusLabel = styled('span')({
    fontWeight: 'bold',
    WebkitFontSmoothing: 'antialiased',
    textTransform: 'uppercase',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
});
var StatusIndicator = (function (_super) {
    __extends(StatusIndicator, _super);
    function StatusIndicator() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StatusIndicator.prototype.render = function () {
        var _a = this.props, _b = _a.emphasis, emphasis = _b === void 0 ? StatusIndicatorEmphasis.High : _b, type = _a.type, icon = _a.icon, label = _a.label, elemProps = __rest(_a, ["emphasis", "type", "icon", "label"]);
        var variant = statusIndicatorStyles.variants[type][emphasis];
        return (React.createElement(Container, __assign({ type: type, emphasis: emphasis }, elemProps),
            icon && (React.createElement(SystemIcon, { colorHover: variant.color, color: variant.color, icon: icon, size: 14, style: { paddingRight: spacing.xxxs } })),
            React.createElement(StatusLabel, null, label)));
    };
    StatusIndicator.Type = StatusIndicatorType;
    StatusIndicator.Emphasis = StatusIndicatorEmphasis;
    return StatusIndicator;
}(React.Component));
export default StatusIndicator;
